<!DOCTYPE html>
<html lang="en-us">
  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile" />
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />

  <!-- Enable responsiveness on mobile devices-->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, maximum-scale=1"
  />

  <title>
     Getting Started &middot; Otic reprogramming 
  </title>

  
  <link rel="canonical" href="http://0.0.0.0:4000/docs/getting_started/" />
  

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css" />
  <link rel="stylesheet" href="/public/css/syntax.css" />
  <link rel="stylesheet" href="/public/css/custom.css" />
  <link rel="stylesheet" href="/public/css/lanyon.css" />
  <link
    rel="stylesheet"
    href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400"
  />

  <!-- stylesheet for icons -->
  <link
    rel="stylesheet"
    href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
  />

  <link rel="shortcut icon" href="/public/favicon.ico" />

  <i class="fas fa-egg"></i>

  <!-- RSS -->
  <link
    rel="alternate"
    type="application/rss+xml"
    title="RSS"
    href="http://0.0.0.0:4000/atom.xml"
  />
</head>


  <body>
    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" />

<!-- Toggleable sidebar -->
<!-- Sidebar icon elements -->

<div class="sidebar" id="sidebar">
  <div class="sidebar-logo">
    <img src="/assets/sidebar_icon.png" />
  </div>

  <div id="contact-list" style="text-align: center">
    
    <a href="https://www.kcl.ac.uk/people/andrea-streit">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-users fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="http://orcid.org/0000-0002-5814-4977">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-align-center fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="mailto:alexthiery1@gmail.com">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-envelope fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-github-alt fa-stack-1x"></i>
      </span>
    </a>
    
  </div>

  <div class="sidebar-item">
    <p>Documentation for the Buzzi et al. 2021 <a href="https://github.com/alexthiery/otic-reprogramming" target="_blank">Github repository</a>.</p>
  </div>


<!-- sidebar tabs -->
  <nav class="sidebar-nav">
    <a class="sidebar-nav-item" href="/">Home</a>

    
    
      
        
          <a class="sidebar-nav-item" href="/docs/quick_start/">Quick Start</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="/docs/smartseq2_alignment/">Smartseq2 alignment</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="/docs/nextflow_configuration/">Nextflow configuration</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="/docs/alignment_ATAC/">ATACseq alignment</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="/docs/alignment_ChIP/">ChIPseq alignment</a>
        
      
    
      
        
          <a class="sidebar-nav-item active" href="/docs/getting_started/">Getting Started</a>
        
      
    


    <!-- <ul>
      
      
        <li class="nav-item top-level ">
          
          <a href="/docs/quick_start/"></a>
          <ul>
            
              <li class="nav-item "><a href="/docs/quick_start/">Quick Start</a></li>
            
              <li class="nav-item "><a href="/docs/smartseq2_alignment/">Smartseq2 alignment</a></li>
            
              <li class="nav-item "><a href="/docs/nextflow_configuration/">Nextflow configuration</a></li>
            
              <li class="nav-item "><a href="/docs/alignment_ATAC/">ATACseq alignment</a></li>
            
              <li class="nav-item "><a href="/docs/alignment_ChIP/">ChIPseq alignment</a></li>
            
              <li class="nav-item current"><a href="/docs/getting_started/">Getting Started</a></li>
            
          </ul>
        </li>
      
    </ul> -->

    <!-- <a class="sidebar-nav-item" href="/_docs/getting_started.md">Getting Started</a> -->
    <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming">GitHub repository</a>
    <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming/archive/master.zip">Download</a>

  <div class="sidebar-item">
    <p>&copy; 2021. All rights reserved.</p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="/" title="Home">Otic reprogramming</a>
            <small>Buzzi et al. 2021</small>
          </h3>
        </div>
      </div>

      <div class="container content"><div class="page">
  <h1 class="page-title">Getting Started</h1>
  <h2 id="data-availability">Data availability</h2>

<p>This repository contains the required code to run the entire alignment and downstream analysis pipeline. For those who would like to re-run the analysis, the following files should be downloaded:</p>

<ul>
  <li>to run the analysis including the alignment, the raw fastq sequencing files can be found <a href="">here</a>.</li>
  <li>to run the downstream analysis from the UMI counts generated from 10x Genomics Cell Ranger are embedded can be found within the repository <a href="&quot;./alignmentOut/cellrangerCounts_renamed&quot;">here</a>.</li>
</ul>

<h2 id="analysis-pre-requisites">Analysis pre-requisites</h2>

<p>The pipeline is run using Nextflow and Docker to ensure reproducibility. The repository can be called directly from GitHub, so to re-run the analysis you just need to install <a href="https://www.nextflow.io/docs/latest/getstarted.html#installation">Nextflow</a> and <a href="https://docs.docker.com/get-docker/">Docker</a>.</p>

<p>If you are wanting to run the downstream analysis interactively outside of Nextflow, you still need to download Docker and you will also need to download this repository.</p>

<h2 id="docker">Docker</h2>

<p>Docker allows us to run our analysis through a container with all required libraries and dependencies. This ensures cross-platform reproducibility of our results.</p>

<p>The docker image used can be found <a href="https://hub.docker.com/r/alexthiery/10x_neural_tube">here</a></p>

<p>We have also included Rstudio within our docker image to allow you to run the downstream analysis interactively if you wish - for details on how to do this click <a href="#interactive-downstream-analysis">here</a>.</p>

<h2 id="nextflow">Nextflow</h2>

<p>You can easily re-run our entire pipeline in Nextflow using the following steps:</p>

<ol>
  <li>Install Nextflow (version &gt;=20.07.1) and Docker</li>
  <li>Download chick genome (<a href="ftp://ftp.ensembl.org/pub/release-94/fasta/gallus_gallus/dna/Gallus_gallus.Gallus_gallus-5.0.dna.toplevel.fa.gz">galgal5</a>)</li>
  <li>For this analysis we used a manually edited GTF file as described in the paper methods. However you can also run the analysis with the GTF from ensembl (<a href="ftp://ftp.ensembl.org/pub/release-94/gtf/gallus_gallus/Gallus_gallus.Gallus_gallus-5.0.94.gtf.gz">galgal5</a>)</li>
  <li>Download raw reads from <a href="">here</a></li>
  <li>Make a sampleInfo.csv file containing the sample names and corresponding paths using this <a href="sampleInfo.csv">template</a></li>
  <li>Run Nextflow using the following command</li>
</ol>

<div class="language-sh highlighter-rouge"><div class="highlight"><pre class="highlight"><code>nextflow run alexthiery/10x_neural_tube <span class="se">\</span>
<span class="nt">-profile</span> docker <span class="se">\</span>
<span class="nt">--metadata</span> &lt;path to sampleInfo.csv&gt; <span class="se">\</span>
<span class="nt">--gtf</span> &lt;path to gtf&gt; <span class="se">\</span>
<span class="nt">--fa</span> &lt;path to genome&gt;
</code></pre></div></div>

<p>This pipeline is configured to be ran on a cluster with 224GB memory and 32CPUs by default. The <code class="language-plaintext highlighter-rouge">-profile</code> flag can be used to set either ‘docker’ or ‘singularity’, depending on the container application installed on your system. These settings can be adjusted by replacing the <code class="language-plaintext highlighter-rouge">-profile</code> flag with a custom config file as below.</p>

<div class="language-sh highlighter-rouge"><div class="highlight"><pre class="highlight"><code>nextflow run alexthiery/10x_neural_tube <span class="se">\</span>
<span class="nt">-c</span> &lt;path to custom.config file&gt; <span class="se">\</span>
<span class="nt">--metadata</span> &lt;path to sampleInfo.csv&gt; <span class="se">\</span>
<span class="nt">--gtf</span> &lt;path to gtf&gt; <span class="se">\</span>
<span class="nt">--fa</span> &lt;path to genome&gt;
</code></pre></div></div>

<p>For a template of a custom.config file, see <a href="conf/crick.config">crick.config</a>. Further information on Nextflow config files can be found <a href="https://www.nextflow.io/docs/latest/config.html#configuration-file">here</a>.</p>

<h2 id="interactive-downstream-analysis">Interactive downstream analysis</h2>

<p>If do not want to re-run the alignment, but would like to run the downstream analysis from the count files, you can run RStudio from within the docker container.</p>

<p>To do this, follow these steps:</p>

<ol>
  <li>clone this repository to your local computer</li>
  <li>start a terminal session and download the docker image - <code class="language-plaintext highlighter-rouge">docker pull alexthiery/10x_neural_tube:v1.0</code></li>
  <li>within terminal launch a docker container interactively - <code class="language-plaintext highlighter-rouge">docker run --rm -e PASSWORD=test -p 8787:8787 -v &lt;PATH_TO_LOCAL_REPOSITORY&gt;:/home/rstudio alexthiery/10x_neural_tube:v1.0</code></li>
  <li>go to <code class="language-plaintext highlighter-rouge">localhost:8787</code> in the browser to open RStudio</li>
</ol>

</div>
</div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="/public/js/script.js"></script>
  </body>
</html>
