<!DOCTYPE html>
<html lang="en-us">
  <head>
  <!-- <script language="JavaScript">
    var password;

    var pass1 = "StreitLab";

    if (localStorage.getItem("pass") != pass1) {
      password = prompt("Please enter your password to view this page!", "");

      password === pass1
        ? localStorage.setItem("pass", password)
        : location.reload();
    }
  </script> -->

  <link href="http://gmpg.org/xfn/11" rel="profile" />
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />

  <!-- Enable responsiveness on mobile devices-->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, maximum-scale=1"
  />

  <title>
     Smartseq2_alignment &middot; Otic reprogramming 
  </title>

  
  <link rel="canonical" href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/general/smartseq2_alignment/" />
  

  <!-- CSS -->
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/poole.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/syntax.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/custom.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/lanyon.css" />
  <link
    rel="stylesheet"
    href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400"
  />

  <!-- stylesheet for icons -->
  <link
    rel="stylesheet"
    href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
  />

  <link rel="shortcut icon" href="https://alexthiery.github.io/otic-reprogramming/public/favicon.ico" />

  <i class="fas fa-egg"></i>

  <!-- RSS -->
  <link
    rel="alternate"
    type="application/rss+xml"
    title="RSS"
    href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/atom.xml"
  />
</head>


  <body>
    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" />

<!-- Toggleable sidebar -->
<!-- Sidebar icon elements -->

<div class="sidebar" id="sidebar">
  <div class="sidebar-logo">
    <img src="https://alexthiery.github.io/otic-reprogramming/assets/sidebar_icon.png" />
  </div>

  <div id="contact-list" style="text-align: center">
    
    <a href="https://www.kcl.ac.uk/people/andrea-streit">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-users fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/otic-reprogramming">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-align-center fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="mailto:alexthiery1@gmail.com">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-envelope fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-github-alt fa-stack-1x"></i>
      </span>
    </a>
    
  </div>

  <div class="sidebar-item">
    <p>Documentation for the Buzzi et al. 2022 <a href="https://github.com/alexthiery/otic-reprogramming" target="_blank">Github repository</a>.</p>
  </div>

  <!-- sidebar tabs -->

  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/">Home</a>

  <!-- add tabs automatically from downstream and general collections-->

     
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/quick_start/"
    >Quick start</a
  >
      
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/smartseq2_alignment/"
    >SmartSeq2 alignment</a
  >
      
  <li class="sidebar-nav-item">
    
    <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Downstream analysis</a>
    <ul>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Intro</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/lmx1a_downstream/">Lmx1aE1 vs Sox3U3 DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/smartseq2_downstream/">SmartSeq2 scRNAseq</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/sox8_downstream/">Sox8OE DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/enhancer_downstream/">Enhancer discovery</a>
      </li>
      
    </ul>
  </li>
  

  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming"
    >GitHub repository</a
  >

  <div class="sidebar-item">
    <p>&copy; 2022. All rights reserved.</p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="https://alexthiery.github.io/otic-reprogramming/" title="Home">Otic reprogramming</a>
            <small>Buzzi et al. 2022</small>
          </h3>
        </div>
      </div>

      <div class="container content"><div class="page">
  <!-- hide auto title generation -->
  <!-- <h1 class="page-title">Smartseq2_alignment</h1> -->
  <h2 id="smartseq2-alignment">SmartSeq2 alignment</h2>
</br>
<p>Our SmartSeq2 single cell RNAseq data is aligned using a custom Nextflow pipeline. This pipeline is built using <a href="https://www.nextflow.io/docs/latest/dsl2.html">Nextflow DSL2</a>, which allows for modularisation and re-usability of each tool.</p>
<p>Each process is configured to run within a separate container, meaning that package versions are hard coded into the pipeline. In order to change these package versions, the container used will need to be changed within the individual process.</p>
<p>Each of the processes used in the pipeline can be found by going to the path in the <code>include {process} from &quot;&lt;PATH&gt;&quot;</code> statements in any of the workflows below.</p>
<p><em>If you would like to re-run the alignment, follow the instructions <a href="https://alexthiery.github.io/otic-reprogramming/general/quick_start#smartseq">here</a>.</em></p>
<hr />
</br>
<h3 id="workflows">Workflows</h3>
<p>Our custom pipeline integrates both alignment and RNA velocity (<a href="http://velocyto.org/">Velocyto</a>).</p>
<p>For simplicity, the pipeline is split into four sub-workflows:</p>
<ul>
<li><a href="#GFP">Modify genome to add GFP sequence</a></li>
<li><a href="#align">Align reads</a></li>
<li><a href="#process_reads">Process readcounts for downstream</a></li>
<li><a href="#velocyto">Run RNA velocity</a></li>
</ul>
</br>
<p>The main workflow integrates each of these sub-workflows.</p>
<details open><summary class="simple">Collapse main workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="n">include</span> <span class="o">{</span><span class="n">add_gfp</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../workflows/add_gfp/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">smartseq2_align</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../workflows/scRNAseq_alignment/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">process_counts</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../workflows/process_counts/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">velocyto_smartseq2</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../workflows/velocyto_smartseq2/main.nf"</span>

<span class="n">Channel</span>
    <span class="o">.</span><span class="na">value</span><span class="o">(</span><span class="n">file</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">fasta</span><span class="o">,</span> <span class="nl">checkIfExists:</span> <span class="kc">true</span><span class="o">))</span>
    <span class="o">.</span><span class="na">set</span> <span class="o">{</span><span class="n">ch_fasta</span><span class="o">}</span>

<span class="n">Channel</span>
    <span class="o">.</span><span class="na">value</span><span class="o">(</span><span class="n">file</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">gtf</span><span class="o">,</span> <span class="nl">checkIfExists:</span> <span class="kc">true</span><span class="o">))</span>
    <span class="o">.</span><span class="na">set</span> <span class="o">{</span><span class="n">ch_gtf</span><span class="o">}</span>

<span class="n">Channel</span>
    <span class="o">.</span><span class="na">value</span><span class="o">(</span><span class="n">file</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">gfp_seq</span><span class="o">,</span> <span class="nl">checkIfExists:</span> <span class="kc">true</span><span class="o">))</span>
    <span class="o">.</span><span class="na">set</span> <span class="o">{</span><span class="n">ch_gfp_seq</span><span class="o">}</span>

<span class="n">workflow</span> <span class="o">{</span>
    <span class="c1">// add gfp to genome and gtf</span>
    <span class="n">add_gfp</span> <span class="o">(</span><span class="n">ch_fasta</span><span class="o">,</span> <span class="n">ch_gtf</span><span class="o">,</span> <span class="n">ch_gfp_seq</span><span class="o">)</span>

    <span class="c1">// align using smartseq2 workflow</span>
    <span class="n">smartseq2_align</span> <span class="o">(</span><span class="n">add_gfp</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">genome</span><span class="o">,</span> <span class="n">add_gfp</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">gtf</span><span class="o">,</span> <span class="n">params</span><span class="o">.</span><span class="na">input</span><span class="o">)</span>

    <span class="c1">// merge and extract gfp counts</span>
    <span class="n">process_counts</span> <span class="o">(</span><span class="n">smartseq2_align</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">htseq_count_files</span><span class="o">)</span>

    <span class="c1">// run velocyto</span>
    <span class="n">velocyto_smartseq2</span> <span class="o">(</span><span class="n">smartseq2_align</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">bam_files</span><span class="o">,</span> <span class="n">ch_gtf</span><span class="o">)</span>

    <span class="c1">// view output</span>
    <span class="n">process_counts</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">processed_counts</span> <span class="o">|</span> <span class="n">view</span>
    <span class="n">velocyto_smartseq2</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">velocyto_counts</span> <span class="o">|</span> <span class="n">view</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
<hr />
</br>
<p><strong><em>Modify genome to add GFP sequence</em></strong><a name="GFP"></a></p>
<p>This sub-workflow takes a GFP sequence from the base repository and appends the genome provided in order to determine the number of GFP reads downstream.</p>
<details open><summary class="simple">Collapse sub-workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Module inclusions
--------------------------------------------------------------------------------------*/</span>
<span class="n">include</span> <span class="o">{</span><span class="n">add_genome_gfp</span><span class="o">;</span> <span class="n">add_gtf_gfp</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../modules/genome-tools/main.nf"</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Define sub workflow
--------------------------------------------------------------------------------------*/</span>

<span class="n">workflow</span> <span class="n">add_gfp</span> <span class="o">{</span>
    <span class="nl">take:</span>
        <span class="n">genome</span>
        <span class="n">gtf</span>
        <span class="n">gfp_seq</span>

    <span class="nl">main:</span>
        <span class="n">add_genome_gfp</span> <span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'add_genome_gfp'</span><span class="o">],</span> <span class="n">genome</span><span class="o">,</span> <span class="n">gfp_seq</span><span class="o">)</span>
        <span class="n">add_gtf_gfp</span> <span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'add_gtf_gfp'</span><span class="o">],</span> <span class="n">gtf</span><span class="o">,</span> <span class="n">gfp_seq</span><span class="o">)</span>

    <span class="nl">emit:</span>
        <span class="n">genome</span> <span class="o">=</span> <span class="n">add_genome_gfp</span><span class="o">.</span><span class="na">out</span>
        <span class="n">gtf</span> <span class="o">=</span> <span class="n">add_gtf_gfp</span><span class="o">.</span><span class="na">out</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
<hr />
</br>
<p><strong><em>Align reads</em></strong><a name="align"></a></p>
<p>First we trim adaptor sequences using Cutadapt v2.10. We then align reads to GalGal6 with HISAT2 v2.2.1. Samtools v1.10 is used to convert SAM to BAM, before finally obtaining gene counts with HTSeq v0.12.4.</p>
<details open><summary class="simple">Collapse sub-workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Module inclusions
--------------------------------------------------------------------------------------*/</span>
<span class="n">include</span> <span class="o">{</span><span class="n">smartseq2_fastq_metadata</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/metadata/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">cutadapt</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/cutadapt/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">hisat2_build</span><span class="o">;</span> <span class="n">hisat2_splice_sites</span><span class="o">;</span> <span class="n">hisat2_splice_align</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/hisat2/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">samtools_view</span> <span class="k">as</span> <span class="n">samtools_view_a</span><span class="o">;</span><span class="n">samtools_view</span> <span class="k">as</span> <span class="n">samtools_view_b</span><span class="o">;</span> <span class="n">samtools_sort</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/samtools/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">htseq_count</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/htseq/main.nf"</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Define sub workflow
--------------------------------------------------------------------------------------*/</span>

<span class="n">workflow</span> <span class="n">smartseq2_align</span> <span class="o">{</span>
    <span class="nl">take:</span>
        <span class="n">genome</span>
        <span class="n">gtf</span>
        <span class="n">sample_csv</span>

    <span class="nl">main:</span>
        <span class="n">smartseq2_fastq_metadata</span> <span class="o">(</span><span class="n">sample_csv</span><span class="o">)</span>

        <span class="n">cutadapt</span> <span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'cutadapt'</span><span class="o">],</span> <span class="n">smartseq2_fastq_metadata</span><span class="o">.</span><span class="na">out</span><span class="o">)</span>
        <span class="n">hisat2_build</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'hisat2_build'</span><span class="o">],</span> <span class="n">genome</span> <span class="o">)</span>
        <span class="n">hisat2_splice_sites</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'hisat2_splice_sites'</span><span class="o">],</span> <span class="n">gtf</span> <span class="o">)</span>
        <span class="n">hisat2_splice_align</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'hisat2_splice_align'</span><span class="o">],</span> <span class="n">cutadapt</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">fastq</span><span class="o">,</span> <span class="n">hisat2_build</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">genome_index</span><span class="o">.</span><span class="na">collect</span><span class="o">(),</span> <span class="n">hisat2_splice_sites</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">splice_sites</span><span class="o">.</span><span class="na">collect</span><span class="o">()</span> <span class="o">)</span>

        <span class="n">samtools_view_a</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'samtools_view_a'</span><span class="o">],</span> <span class="n">hisat2_splice_align</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">sam</span> <span class="o">)</span>
        <span class="n">samtools_sort</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'samtools_sort'</span><span class="o">],</span> <span class="n">samtools_view_a</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">bam</span> <span class="o">)</span>
        <span class="n">samtools_view_b</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'samtools_view_b'</span><span class="o">],</span> <span class="n">samtools_sort</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">bam</span> <span class="o">)</span>

        <span class="n">htseq_count</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'htseq_count'</span><span class="o">],</span> <span class="n">samtools_view_b</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">bam</span><span class="o">,</span> <span class="n">gtf</span> <span class="o">)</span>

    <span class="nl">emit:</span>
        <span class="n">bam_files</span> <span class="o">=</span> <span class="n">samtools_sort</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">bam</span>
        <span class="n">htseq_count_files</span> <span class="o">=</span> <span class="n">htseq_count</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">counts</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
<hr />
</br>
<p><strong><em>Process read counts for input into Antler</em></strong><a name="process_reads"></a></p>
<p>The merge counts process is a custom R script which formats cell names and generates the phenoData.csv, assayData.csv and gfpData.csv required in our downstream pipeline.</p>
<details open><summary class="simple">Collapse sub-workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Module inclusions
--------------------------------------------------------------------------------------*/</span>
<span class="n">include</span> <span class="o">{</span><span class="n">r_analysis</span> <span class="k">as</span> <span class="n">merge_counts</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../modules/r_analysis/main.nf"</span>


<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Define sub workflow
--------------------------------------------------------------------------------------*/</span>

<span class="n">workflow</span> <span class="n">process_counts</span> <span class="o">{</span>
    <span class="nl">take:</span>
        <span class="n">count_files</span>

    <span class="nl">main:</span>
        <span class="c1">// group counts into a single channel for merge cell counts</span>
        <span class="n">ch_all_counts</span> <span class="o">=</span> <span class="n">count_files</span>
            <span class="o">.</span><span class="na">map</span> <span class="o">{</span> <span class="o">[</span><span class="n">file</span><span class="o">(</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">],</span> <span class="nl">checkIfExists:</span> <span class="kc">true</span><span class="o">)]</span> <span class="o">}</span>
            <span class="o">.</span><span class="na">collect</span><span class="o">()</span>

        <span class="c1">// merge cell counts into csv</span>
        <span class="n">merge_counts</span> <span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'merge_counts'</span><span class="o">],</span> <span class="n">ch_all_counts</span><span class="o">)</span>

    <span class="nl">emit:</span>
        <span class="n">processed_counts</span> <span class="o">=</span> <span class="n">merge_counts</span><span class="o">.</span><span class="na">out</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
</br>
<details><summary class="simple">Collapse merge_counts.R</summary>
<p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#!/usr/bin/env Rscript</span><span class="w">

</span><span class="c1"># Define arguments for Rscript</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">getopt</span><span class="p">)</span><span class="w">
</span><span class="n">spec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="w">
  </span><span class="s1">'runtype'</span><span class="p">,</span><span class="w"> </span><span class="s1">'l'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"character"</span><span class="p">,</span><span class="w">
  </span><span class="s1">'cores'</span><span class="w">   </span><span class="p">,</span><span class="w"> </span><span class="s1">'c'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"integer"</span><span class="w">
</span><span class="p">),</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="n">opt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">getopt</span><span class="p">(</span><span class="n">spec</span><span class="p">)</span><span class="w">

</span><span class="c1"># Set run location</span><span class="w">
</span><span class="k">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">){</span><span class="w">
  </span><span class="n">cat</span><span class="p">(</span><span class="s1">'No command line arguments provided, user defaults paths are set for running interactively in Rstudio on docker\n'</span><span class="p">)</span><span class="w">
  </span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"user"</span><span class="w">
</span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"user"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># Set paths and load data</span><span class="w">
</span><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"user"</span><span class="p">){</span><span class="w">
    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/merge_counts/output/"</span><span class="w">
    </span><span class="n">input_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">list.files</span><span class="p">(</span><span class="s2">"./testData/process_counts/"</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"*.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

  </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">cat</span><span class="p">(</span><span class="s1">'pipeline running through nextflow\n'</span><span class="p">)</span><span class="w">

    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"output/"</span><span class="w">
    </span><span class="n">input_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">list.files</span><span class="p">(</span><span class="s2">"./"</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"*.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

  </span><span class="n">library</span><span class="p">(</span><span class="n">plyr</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="n">assayData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">do.call</span><span class="p">(</span><span class="n">cbind</span><span class="p">,</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">input_files</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">f</span><span class="p">){</span><span class="w">
	</span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">f</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w">
	</span><span class="n">df</span><span class="p">[</span><span class="n">seq</span><span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="m">-5</span><span class="p">),,</span><span class="w"> </span><span class="n">drop</span><span class="o">=</span><span class="nb">F</span><span class="p">]</span><span class="w">
</span><span class="p">}))</span><span class="w">

</span><span class="c1"># ss11 ss15 WTCHG_528508_201189.txt -&gt; 201189</span><span class="w">
</span><span class="c1"># ss89 TSS_P2_E2_13-1234.txt -&gt; P2E2</span><span class="w">
</span><span class="n">phenoData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="n">ldply</span><span class="p">(</span><span class="n">basename</span><span class="p">(</span><span class="n">input_files</span><span class="p">),</span><span class="w"> </span><span class="n">.fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"ss11"</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">))</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nf">c</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">tools</span><span class="o">::</span><span class="n">file_path_sans_ext</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"_"</span><span class="p">)[[</span><span class="m">1</span><span class="p">]][[</span><span class="m">3</span><span class="p">]],</span><span class="w"> </span><span class="m">11</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s2">"sc"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w">  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"ss15"</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">))</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nf">c</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">tools</span><span class="o">::</span><span class="n">file_path_sans_ext</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"_"</span><span class="p">)[[</span><span class="m">1</span><span class="p">]][[</span><span class="m">3</span><span class="p">]],</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s2">"sc"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="n">spl1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">strsplit</span><span class="p">(</span><span class="n">tools</span><span class="o">::</span><span class="n">file_path_sans_ext</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"_"</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]</span><span class="w">
    </span><span class="nf">c</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">spl1</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">strsplit</span><span class="p">(</span><span class="n">spl1</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-"</span><span class="p">)[[</span><span class="m">1</span><span class="p">]][</span><span class="m">1</span><span class="p">]),</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s2">"sc"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
</span><span class="p">}),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"cell_ID"</span><span class="p">,</span><span class="w"> </span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="s2">"replicate_id"</span><span class="p">,</span><span class="w"> </span><span class="s2">"treatment"</span><span class="p">))</span><span class="w">

</span><span class="n">rownames</span><span class="p">(</span><span class="n">phenoData</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">phenoData</span><span class="o">$</span><span class="n">cell_ID</span><span class="w">
</span><span class="n">phenoData</span><span class="o">$</span><span class="n">cell_ID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">assayData</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">phenoData</span><span class="p">)</span><span class="w">

</span><span class="c1"># make dataframe for gfp counts</span><span class="w">
</span><span class="n">gfpData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">assayData</span><span class="p">[</span><span class="s2">"GFP"</span><span class="p">,]</span><span class="w">

</span><span class="c1"># remove gfp counts from assayData</span><span class="w">
</span><span class="n">assayData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">assayData</span><span class="p">[</span><span class="o">-</span><span class="n">which</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">assayData</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"GFP"</span><span class="p">),]</span><span class="w">

</span><span class="n">write.table</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">gfpData</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'gfpData.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="kc">NA</span><span class="p">)</span><span class="w">
</span><span class="n">write.table</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">assayData</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'assayData.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="kc">NA</span><span class="p">)</span><span class="w">
</span><span class="n">write.table</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">phenoData</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'phenoData.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s1">'\t'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="kc">NA</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</details>
<hr />
</br>
<p><strong><em>Run RNA velocity</em></strong><a name="velocyto"></a></p>
<p>The RNA velocity sub-workflow takes the output BAM files from HISAT2 and groups all the cells into a single channel input. We then run Velocyto which generates a single <code>.loom</code> file output containing spliced, unspliced and spanning reads.</p>
<details open><summary class="simple">Collapse sub-workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Module inclusions
--------------------------------------------------------------------------------------*/</span>
<span class="n">include</span> <span class="o">{</span><span class="n">velocyto_run_smartseq2</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/velocyto/main.nf"</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Define sub workflow
--------------------------------------------------------------------------------------*/</span>

<span class="n">workflow</span> <span class="n">velocyto_smartseq2</span> <span class="o">{</span>
    <span class="nl">take:</span>
        <span class="n">bam_files</span>
        <span class="n">gtf</span>

    <span class="nl">main:</span>
        <span class="c1">// group bams into a single channel for velocyto</span>
        <span class="n">ch_velocyto_bam</span> <span class="o">=</span> <span class="n">bam_files</span>
            <span class="o">.</span><span class="na">map</span> <span class="o">{</span> <span class="o">[[</span><span class="nl">sample_id:</span><span class="s2">"all_cells"</span><span class="o">],</span> <span class="n">file</span><span class="o">(</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">],</span> <span class="nl">checkIfExists:</span> <span class="kc">true</span><span class="o">)]</span> <span class="o">}</span>
            <span class="o">.</span><span class="na">groupTuple</span><span class="o">(</span><span class="nl">by:</span> <span class="mi">0</span><span class="o">)</span>

        <span class="n">velocyto_run_smartseq2</span> <span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'velocyto_run_smartseq2'</span><span class="o">],</span> <span class="n">ch_velocyto_bam</span><span class="o">,</span> <span class="n">gtf</span> <span class="o">)</span>

    <span class="nl">emit:</span>
        <span class="n">velocyto_counts</span> <span class="o">=</span> <span class="n">velocyto_run_smartseq2</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">velocyto</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
</br>

</div>
</div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="https://alexthiery.github.io/otic-reprogramming/public/js/script.js"></script>
  </body>
</html>
