<!DOCTYPE html>
<html lang="en-us">
  <head>
  <script language="JavaScript">
    var password;

    var pass1 = "StreitLab";

    if (localStorage.getItem("pass") != pass1) {
      password = prompt("Please enter your password to view this page!", "");

      password === pass1
        ? localStorage.setItem("pass", password)
        : location.reload();
    }
  </script>

  <link href="http://gmpg.org/xfn/11" rel="profile" />
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />

  <!-- Enable responsiveness on mobile devices-->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, maximum-scale=1"
  />

  <title>
     Enhancer_downstream &middot; Otic reprogramming 
  </title>

  
  <link rel="canonical" href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/downstream/enhancer_downstream/" />
  

  <!-- CSS -->
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/poole.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/syntax.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/custom.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/lanyon.css" />
  <link
    rel="stylesheet"
    href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400"
  />

  <!-- stylesheet for icons -->
  <link
    rel="stylesheet"
    href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
  />

  <link rel="shortcut icon" href="https://alexthiery.github.io/otic-reprogramming/public/favicon.ico" />

  <i class="fas fa-egg"></i>

  <!-- RSS -->
  <link
    rel="alternate"
    type="application/rss+xml"
    title="RSS"
    href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/atom.xml"
  />
</head>


  <body>
    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" />

<!-- Toggleable sidebar -->
<!-- Sidebar icon elements -->

<div class="sidebar" id="sidebar">
  <div class="sidebar-logo">
    <img src="https://alexthiery.github.io/otic-reprogramming/assets/sidebar_icon.png" />
  </div>

  <div id="contact-list" style="text-align: center">
    
    <a href="https://www.kcl.ac.uk/people/andrea-streit">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-users fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/otic-reprogramming">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-align-center fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="mailto:alexthiery1@gmail.com">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-envelope fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-github-alt fa-stack-1x"></i>
      </span>
    </a>
    
  </div>

  <div class="sidebar-item">
    <p>Documentation for the Buzzi et al. 2021 <a href="https://github.com/alexthiery/otic-reprogramming" target="_blank">Github repository</a>.</p>
  </div>

  <!-- sidebar tabs -->

  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/">Home</a>

  <!-- add tabs automatically from downstream and general collections-->

     
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/quick_start/"
    >Quick start</a
  >
      
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/smartseq2_alignment/"
    >SmartSeq2 alignment</a
  >
      
  <li class="sidebar-nav-item">
    
    <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Downstream analysis</a>
    <ul>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Intro</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/lmx1a_downstream/">Lmx1aE1 vs Sox3U3 DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/smartseq2_downstream/">SmartSeq2 scRNAseq</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/sox8_downstream/">Sox8OE DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/enhancer_downstream/">Enhancer discovery</a>
      </li>
      
    </ul>
  </li>
  

  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming"
    >GitHub repository</a
  >
  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming/archive/master.zip"
    >Download</a
  >

  <div class="sidebar-item">
    <p>&copy; 2021. All rights reserved.</p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="https://alexthiery.github.io/otic-reprogramming/" title="Home">Otic reprogramming</a>
            <small>Buzzi et al. 2021</small>
          </h3>
        </div>
      </div>

      <div class="container content"><div class="page">
  <!-- hide auto title generation -->
  <!-- <h1 class="page-title">Enhancer_downstream</h1> -->
  <h2 id="enhancer-discovery">Enhancer discovery</h2>
</br>
<p>We performed epigenomic profiling carrying out ChIPseq for H3K27ac, H3K27me3 and H3K4me3 as well as ATACseq of OEP dissected tissue. Below we describe the enhancer discovery pipeline.</p>
<p>To view our cloned enhancer tracks, click <a href="#enhancer_visualisation">here</a>.</p>
<hr />
</br>
<p>Our enhancer discovery pipeline integrates ChIP (H3K27Ac and H3K27me3) and ATAC data into a single Nextflow sub-workflow. This sub-workflow is executed as part of the downstream analysis pipeline. <strong>If you would like to re-run the downstream analysis pipeline, follow the instructions <a href="https://alexthiery.github.io/otic-reprogramming/general/quick_start#downstream">here</a>.</strong></p>
<hr />
</br>
<h3 id="sub-workflow">Sub-workflow</h3>
</br>
<p>The sub-workflow uses Bedtools and Homer in order to identify putative enhancers:</p>
<ul>
<li>First, we subset protein coding genes from our GTF.</li>
<li>Next, we use Bedtools to subset ATAC+H3K27Ac-H3K27me3 peaks.</li>
<li>Finally, we annotate the remaining peaks to the nearest protein coding gene before removing peaks which fall within promoter or exonic regions, as they are unlikely to be active enhancers.</li>
</ul>
<p><a href="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/putative_enhancers/putative_enhancers.bed" download>Download putative enhancer co-ordinates</a></p>
<p><a href="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/putative_enhancers/putative_enhancers_annotated.txt download">Download annotated putative enhancers</a></p>
</br>
<p>The sub-workflow also integrates custom R scripts for:</p>
<ul>
<li><a href="#annotation_frequency">Plotting peak annotation frequency</a></li>
<li><a href="#functional_enrichment">Functional enrichment analysis</a></li>
<li><a href="#enhancer_profile">Plotting putative enhancer profile</a></li>
<li><a href="#motif_enrichment">Motif enrichment analysis from Homer output</a></li>
</ul>
<details open><summary class="simple">Collapse enhancer discovery sub-workflow</summary>
<p>
<div class="language-groovy highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">// Define DSL2</span>
<span class="n">nextflow</span><span class="o">.</span><span class="na">enable</span><span class="o">.</span><span class="na">dsl</span><span class="o">=</span><span class="mi">2</span>

<span class="n">include</span> <span class="o">{</span><span class="n">awk</span> <span class="k">as</span> <span class="n">awk_enhancer_filter</span><span class="o">;</span> <span class="n">awk</span> <span class="k">as</span> <span class="n">awk_gtf_filter</span><span class="o">;</span> <span class="n">awk</span> <span class="k">as</span> <span class="n">awk_trim_head</span><span class="o">;</span> <span class="n">cut</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/luslab_linux_tools/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">bedtools_intersect</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/bedtools/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">bedtools_subtract</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/bedtools/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">homer_annotate_peaks</span><span class="o">;</span> <span class="n">homer_find_motifs</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../luslab-nf-modules/tools/homer/main.nf"</span>
<span class="n">include</span> <span class="o">{</span><span class="n">r_analysis</span> <span class="k">as</span> <span class="n">enhancer_profile</span><span class="o">;</span> <span class="n">r_analysis</span> <span class="k">as</span> <span class="n">plot_motifs</span><span class="o">;</span> <span class="n">r_analysis</span> <span class="k">as</span> <span class="n">functional_enrichment_analysis</span><span class="o">;</span> <span class="n">r_analysis</span> <span class="k">as</span> <span class="n">peak_annotations_frequency</span><span class="o">}</span> <span class="n">from</span> <span class="s2">"$baseDir/../modules/r_analysis/main.nf"</span>

<span class="cm">/*------------------------------------------------------------------------------------*/</span>
<span class="cm">/* Define sub workflow
--------------------------------------------------------------------------------------*/</span>

<span class="n">workflow</span> <span class="n">enhancer_analysis</span> <span class="o">{</span>
    <span class="nl">take:</span>
        <span class="n">chip_bigwig</span>
        <span class="n">atac_bigwig</span>
        <span class="n">chip_peaks</span>
        <span class="n">atac_peaks</span>
        <span class="n">genome</span>
        <span class="n">gtf</span>

    <span class="nl">main:</span>
        <span class="c1">// Keep only protein coding genes in gtf</span>
        <span class="n">awk_gtf_filter</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'awk_gtf_filter'</span><span class="o">],</span> <span class="n">gtf</span><span class="o">)</span>

        <span class="c1">// // Intersect ATAC peaks with H3K27Ac peaks</span>
        <span class="n">bedtools_intersect</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'bedtools_intersect'</span><span class="o">],</span> <span class="n">atac_peaks</span><span class="o">.</span><span class="na">filter</span><span class="o">{</span> <span class="n">it</span><span class="o">[</span><span class="mi">0</span><span class="o">].</span><span class="na">sample_id</span> <span class="o">==</span> <span class="s1">'ATAC'</span> <span class="o">},</span> <span class="n">chip_peaks</span><span class="o">.</span><span class="na">filter</span><span class="o">{</span> <span class="n">it</span><span class="o">[</span><span class="mi">0</span><span class="o">].</span><span class="na">sample_id</span> <span class="o">==</span> <span class="s1">'H3K27Ac'</span> <span class="o">}.</span><span class="na">map</span><span class="o">{</span> <span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]</span> <span class="o">}</span> <span class="o">)</span>

        <span class="c1">// Remove any peaks which also have hits for H3K27me3</span>
        <span class="n">bedtools_subtract</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'bedtools_subtract'</span><span class="o">],</span> <span class="n">bedtools_intersect</span><span class="o">.</span><span class="na">out</span><span class="o">,</span> <span class="n">chip_peaks</span><span class="o">.</span><span class="na">filter</span><span class="o">{</span> <span class="n">it</span><span class="o">[</span><span class="mi">0</span><span class="o">].</span><span class="na">sample_id</span> <span class="o">==</span> <span class="s1">'H3K27me3'</span> <span class="o">}.</span><span class="na">map</span><span class="o">{</span> <span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]</span> <span class="o">}</span> <span class="o">)</span>

        <span class="c1">// Annotate remaining peaks</span>
        <span class="n">homer_annotate_peaks</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'homer_annotate_peaks'</span><span class="o">],</span> <span class="n">bedtools_subtract</span><span class="o">.</span><span class="na">out</span><span class="o">,</span> <span class="n">genome</span><span class="o">,</span> <span class="n">awk_gtf_filter</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file_no_meta</span><span class="o">)</span>

        <span class="c1">// Plot distribution of intersected peak annotations</span>
        <span class="n">peak_annotations_frequency</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'peak_annotations_frequency'</span><span class="o">],</span> <span class="n">homer_annotate_peaks</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]})</span>

        <span class="c1">// Remove peaks in protein coding promoters and  protein coding exons</span>
        <span class="n">awk_enhancer_filter</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'awk_enhancer_filter'</span><span class="o">],</span> <span class="n">homer_annotate_peaks</span><span class="o">.</span><span class="na">out</span><span class="o">)</span>

        <span class="c1">// Run functional enrichment analysis on annotated putative enhancers</span>
        <span class="n">functional_enrichment_analysis</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'functional_enrichment_analysis'</span><span class="o">],</span> <span class="n">awk_enhancer_filter</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file_no_meta</span><span class="o">)</span>

        <span class="c1">// Combine all peak data for enhancer profile input</span>
        <span class="n">enhancer_profile_input</span> <span class="o">=</span> <span class="n">chip_peaks</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]}.</span><span class="na">flatten</span><span class="o">().</span><span class="na">collect</span><span class="o">()</span>
            <span class="o">.</span><span class="na">combine</span><span class="o">(</span><span class="n">chip_bigwig</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]}.</span><span class="na">flatten</span><span class="o">().</span><span class="na">collect</span><span class="o">())</span>
            <span class="o">.</span><span class="na">combine</span><span class="o">(</span><span class="n">atac_peaks</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]}.</span><span class="na">flatten</span><span class="o">().</span><span class="na">collect</span><span class="o">())</span>
            <span class="o">.</span><span class="na">combine</span><span class="o">(</span><span class="n">atac_bigwig</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]}.</span><span class="na">flatten</span><span class="o">().</span><span class="na">collect</span><span class="o">())</span>
            <span class="o">.</span><span class="na">combine</span><span class="o">(</span><span class="n">awk_enhancer_filter</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file_no_meta</span><span class="o">)</span>

        <span class="c1">// Plot ChIP and ATAC profile across enhancers</span>
        <span class="n">enhancer_profile</span><span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'enhancer_profile'</span><span class="o">],</span> <span class="n">enhancer_profile_input</span><span class="o">)</span>

        <span class="c1">// Convert awk output to bed file</span>
        <span class="n">cut</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'cut'</span><span class="o">],</span> <span class="n">awk_enhancer_filter</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file</span><span class="o">)</span>

        <span class="c1">// Trim first line from bed file</span>
        <span class="n">awk_trim_head</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'awk_trim_head'</span><span class="o">],</span> <span class="n">cut</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file</span><span class="o">)</span>

        <span class="c1">// Run motif enrichment analysis on remaining peaks</span>
        <span class="n">homer_find_motifs</span><span class="o">(</span><span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'homer_find_motifs'</span><span class="o">],</span> <span class="n">awk_enhancer_filter</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">file</span><span class="o">,</span> <span class="n">genome</span><span class="o">)</span>

        <span class="c1">// Generate motif plot</span>
        <span class="n">plot_motifs</span><span class="o">(</span> <span class="n">params</span><span class="o">.</span><span class="na">modules</span><span class="o">[</span><span class="s1">'plot_motifs'</span><span class="o">],</span> <span class="n">homer_find_motifs</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">enrichedMotifs</span><span class="o">.</span><span class="na">map</span><span class="o">{</span><span class="n">it</span><span class="o">[</span><span class="mi">1</span><span class="o">]}</span> <span class="o">)</span>
<span class="o">}</span>
</code></pre></div></div>
</details>
<hr />
<br/>
<h3 id="plotting-peak-annotation-frequencya-nameannotationfrequencya">Plotting peak annotation frequency<a name="annotation_frequency"></a></h3>
</br>
<details><summary class="box">Code</summary>
<p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">

</span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/'</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">)</span><span class="w">

</span><span class="c1"># import ATAC peaks intersected with +K27Ac -K27me3</span><span class="w">
</span><span class="n">peaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s1">'./'</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="o">=</span><span class="s2">".txt"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">)</span><span class="w">

</span><span class="c1"># extract and simplify annotations for categorisation</span><span class="w">
</span><span class="n">annotation_peaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">sub</span><span class="p">(</span><span class="s1">' .*'</span><span class="p">,</span><span class="w"> </span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">peaks</span><span class="p">[,</span><span class="s2">"Annotation"</span><span class="p">]))</span><span class="w">

</span><span class="c1"># order frequency</span><span class="w">
</span><span class="n">freq_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">prop.table</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">annotation_peaks</span><span class="p">))[</span><span class="n">order</span><span class="p">(</span><span class="n">prop.table</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">annotation_peaks</span><span class="p">)))])</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">freq_data</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'peaks'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Frequency'</span><span class="p">)</span><span class="w">


</span><span class="c1"># plot frequency plot of peak annotations</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"peak_annotation_frequency.png"</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">freq_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">peaks</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Frequency</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="s1">'steelblue'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.title.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.ticks.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.line.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="o">=</span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
</details>
</br>
<img class="myImages width_50" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/peak_annotations_frequency/output/peak_annotation_frequency.png">
<hr />
<br/>
<h3 id="functional-enrichment-analysisa-namefunctionalenrichmenta">Functional enrichment analysis<a name="functional_enrichment"></a></h3>
</br>
<p>To view all of our functional enrichment analysis output in the browser, click <a href="https://biit.cs.ut.ee/gplink/l/IgcsqI0XTi">here</a>.</p>
<details><summary class="box">Code</summary>
<p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">gprofiler2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">

</span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/'</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="n">putative_enhancers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'*txt'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">

</span><span class="c1"># run functional enrichment analysis using GO:biological process and KEGG terms</span><span class="w">
</span><span class="n">fea_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gost</span><span class="p">(</span><span class="n">putative_enhancers</span><span class="o">$</span><span class="n">Entrez.ID</span><span class="p">,</span><span class="w"> </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ggallus'</span><span class="p">,</span><span class="w"> </span><span class="n">sources</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'GO:BP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'KEGG'</span><span class="p">))</span><span class="w">

</span><span class="c1"># generate URL for full results</span><span class="w">
</span><span class="c1"># gost(putative_enhancers$Entrez.ID, organism = 'ggallus', sources = c('GO:BP', 'KEGG'), as_short_link = TRUE)</span><span class="w">

</span><span class="n">go_terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"GO:0007399"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KEGG:04310"</span><span class="p">,</span><span class="w"> </span><span class="s2">"GO:0048839"</span><span class="p">,</span><span class="w"> </span><span class="s2">"GO:0050767"</span><span class="p">,</span><span class="w"> </span><span class="s2">"GO:0043408"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KEGG:04330"</span><span class="p">)</span><span class="w">

</span><span class="c1"># select enriched terms of interest and generate bar plot</span><span class="w">
</span><span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fea_res</span><span class="o">$</span><span class="n">result</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">filter</span><span class="p">(</span><span class="n">term_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">go_terms</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">select</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="n">term_name</span><span class="p">,</span><span class="w"> </span><span class="n">term_id</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">p_value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">arrange</span><span class="p">(</span><span class="n">desc</span><span class="p">(</span><span class="n">`-log10(p_value)`</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="n">term_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">term_name</span><span class="p">,</span><span class="w"> </span><span class="s1">' ('</span><span class="p">,</span><span class="w"> </span><span class="n">term_id</span><span class="p">,</span><span class="w"> </span><span class="s2">")"</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="n">term_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">term_name</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">term_name</span><span class="p">))</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"functional_enrichment.png"</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">term_name</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">p_value</span><span class="p">),</span><span class="w"> </span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">term_name</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="s1">'steelblue'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_text</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'left'</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-2</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.text.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.ticks.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.line.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">26</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.margin</span><span class="o">=</span><span class="n">unit</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="m">0.5</span><span class="p">,</span><span class="m">0.5</span><span class="p">,</span><span class="m">0.5</span><span class="p">),</span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
</details>
</br>
<img class="myImages width_50" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/functional_enrichment_analysis/output/functional_enrichment.png">
<hr />
<br/>
<h3 id="plotting-putative-enhancer-profilea-nameenhancerprofilea">Plotting putative enhancer profile<a name="enhancer_profile"></a></h3>
</br>
<details><summary class="box">Code</summary>
<p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ChIPpeakAnno</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">rtracklayer</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">

</span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/"</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="c1"># import putative enhancer peaks (ATAC peaks; + K27ac; - K27me3; - &lt;2kb upstream TSS; - exons)</span><span class="w">
</span><span class="n">shared.peaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="o">=</span><span class="s2">"*.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">)</span><span class="w">

</span><span class="n">peaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GRanges</span><span class="p">(</span><span class="n">seqnames</span><span class="o">=</span><span class="n">shared.peaks</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="w">
                 </span><span class="n">ranges</span><span class="o">=</span><span class="n">IRanges</span><span class="p">(</span><span class="n">start</span><span class="o">=</span><span class="n">shared.peaks</span><span class="p">[,</span><span class="m">3</span><span class="p">],</span><span class="w">
                                </span><span class="n">end</span><span class="o">=</span><span class="n">shared.peaks</span><span class="p">[,</span><span class="m">4</span><span class="p">],</span><span class="w">
                                </span><span class="n">names</span><span class="o">=</span><span class="n">shared.peaks</span><span class="p">[,</span><span class="m">1</span><span class="p">]))</span><span class="w">

</span><span class="c1"># find centre of ATAC peak and get coordinates for +/-2kb</span><span class="w">
</span><span class="n">peaks.recentered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">peaks.center</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">peaks</span><span class="w">
</span><span class="n">start</span><span class="p">(</span><span class="n">peaks.center</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">start</span><span class="p">(</span><span class="n">peaks</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">floor</span><span class="p">(</span><span class="n">width</span><span class="p">(</span><span class="n">peaks</span><span class="p">)</span><span class="o">/</span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">width</span><span class="p">(</span><span class="n">peaks.center</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">start</span><span class="p">(</span><span class="n">peaks.recentered</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">start</span><span class="p">(</span><span class="n">peaks.center</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">2000</span><span class="w">
</span><span class="n">end</span><span class="p">(</span><span class="n">peaks.recentered</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">end</span><span class="p">(</span><span class="n">peaks.center</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">2000</span><span class="w">


</span><span class="c1"># import bigwig files and select regions corresponding to ATAC (putative enhancer) peaks</span><span class="w">
</span><span class="n">bigwig_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">list.files</span><span class="p">(</span><span class="s1">'./'</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'bigWig'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
</span><span class="n">ATAC.bw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">import</span><span class="p">(</span><span class="n">bigwig_files</span><span class="p">[</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"ATAC"</span><span class="p">,</span><span class="w"> </span><span class="n">bigwig_files</span><span class="p">)],</span><span class="w"> </span><span class="n">format</span><span class="o">=</span><span class="s2">"BigWig"</span><span class="p">,</span><span class="w"> </span><span class="n">which</span><span class="o">=</span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">as</span><span class="o">=</span><span class="s2">"RleList"</span><span class="p">)</span><span class="w">
</span><span class="n">H3K27Ac.bw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">import</span><span class="p">(</span><span class="n">bigwig_files</span><span class="p">[</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"H3K27Ac"</span><span class="p">,</span><span class="w"> </span><span class="n">bigwig_files</span><span class="p">)],</span><span class="w"> </span><span class="n">format</span><span class="o">=</span><span class="s2">"BigWig"</span><span class="p">,</span><span class="w"> </span><span class="n">which</span><span class="o">=</span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">as</span><span class="o">=</span><span class="s2">"RleList"</span><span class="p">)</span><span class="w">
</span><span class="n">H3K27me3.bw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">import</span><span class="p">(</span><span class="n">bigwig_files</span><span class="p">[</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"H3K27me3"</span><span class="p">,</span><span class="w"> </span><span class="n">bigwig_files</span><span class="p">)],</span><span class="w"> </span><span class="n">format</span><span class="o">=</span><span class="s2">"BigWig"</span><span class="p">,</span><span class="w"> </span><span class="n">which</span><span class="o">=</span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">as</span><span class="o">=</span><span class="s2">"RleList"</span><span class="p">)</span><span class="w">
</span><span class="n">input.bw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">import</span><span class="p">(</span><span class="n">bigwig_files</span><span class="p">[</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"input"</span><span class="p">,</span><span class="w"> </span><span class="n">bigwig_files</span><span class="p">)],</span><span class="w"> </span><span class="n">format</span><span class="o">=</span><span class="s2">"BigWig"</span><span class="p">,</span><span class="w"> </span><span class="n">which</span><span class="o">=</span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">as</span><span class="o">=</span><span class="s2">"RleList"</span><span class="p">)</span><span class="w">

</span><span class="c1"># make list of bigwig files</span><span class="w">
</span><span class="n">bw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">ATAC</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ATAC.bw</span><span class="p">,</span><span class="w"> </span><span class="n">H3K27Ac</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">H3K27Ac.bw</span><span class="p">,</span><span class="w"> </span><span class="n">H3K27me3</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">H3K27me3.bw</span><span class="p">,</span><span class="w"> </span><span class="n">Input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">input.bw</span><span class="p">)</span><span class="w">

</span><span class="c1"># extract signal for +/-2kb around enhancer peak for visualisation</span><span class="w">
</span><span class="n">sig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">featureAlignedSignal</span><span class="p">(</span><span class="n">bw</span><span class="p">,</span><span class="w"> </span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w">
                            </span><span class="n">upstream</span><span class="o">=</span><span class="m">2000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="o">=</span><span class="m">2000</span><span class="p">)</span><span class="w">


</span><span class="c1"># plot profile around ATAC peaks</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"metaprofile.png"</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">17</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">featureAlignedDistribution</span><span class="p">(</span><span class="n">sig</span><span class="p">,</span><span class="w"> </span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="o">=</span><span class="m">2000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="o">=</span><span class="m">2000</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s2">"l"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot heatmap</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"heatmap.png"</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">featureAlignedHeatmap</span><span class="p">(</span><span class="n">sig</span><span class="p">,</span><span class="w"> </span><span class="n">peaks.recentered</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="o">=</span><span class="m">2000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="o">=</span><span class="m">2000</span><span class="p">,</span><span class="w"> </span><span class="n">upper.extreme</span><span class="o">=</span><span class="m">2.5</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
</details>
<br/>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'enhancer_profile')">Enhancer profile</button>
  <button class="tablinks" onclick="openTab(event, 'enhancer_heatmap')">Enhancer heatmap</button>
</div>
<br/>
<div id="enhancer_profile" class="tabcontent">
  <img class="myImages" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/enhancer_profile/output/metaprofile.png">
</div>
<div id="enhancer_heatmap" class="tabcontent">
  <img class="myImages" id="myImg"  src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/enhancer_profile/output/heatmap.png">
</div>
<hr />
<br/>
<h3 id="motif-enrichment-analysis-from-homer-outputa-namemotifenrichmenta">Motif enrichment analysis from Homer output<a name="motif_enrichment"></a></h3>
</br>
<details><summary class="box">Code</summary>
<p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggseqlogo</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">gridExtra</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">

</span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/"</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">


</span><span class="c1">######## read in data</span><span class="w">
</span><span class="c1"># read in logo data</span><span class="w">
</span><span class="n">motif_logos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">){</span><span class="w">
  </span><span class="n">motif_logos</span><span class="p">[[</span><span class="n">paste</span><span class="p">(</span><span class="n">i</span><span class="p">)]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">read.delim</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s1">'./ATAC_motif_enrichment/knownResults/known'</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s1">'.motif'</span><span class="p">))[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">motif_logos</span><span class="p">[[</span><span class="n">paste</span><span class="p">(</span><span class="n">i</span><span class="p">)]])</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'C'</span><span class="p">,</span><span class="w"> </span><span class="s1">'G'</span><span class="p">,</span><span class="w"> </span><span class="s1">'T'</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># read in motif info</span><span class="w">
</span><span class="n">motif_meta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s1">'./ATAC_motif_enrichment/knownResults.txt'</span><span class="p">))[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="p">)]</span><span class="w">

</span><span class="c1"># strip name</span><span class="w">
</span><span class="n">motif_meta</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sub</span><span class="p">(</span><span class="s2">"\\(.*"</span><span class="p">,</span><span class="w"> </span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">motif_meta</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w">


</span><span class="c1">####### prepare grobs</span><span class="w">
</span><span class="c1"># gene names</span><span class="w">
</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">as_grob</span><span class="p">(</span><span class="o">~</span><span class="n">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">text</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"Gene"</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">font</span><span class="o">=</span><span class="m">2</span><span class="p">)))</span><span class="w">

</span><span class="n">motif_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">motif_meta</span><span class="p">[,</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="n">as_grob</span><span class="p">(</span><span class="o">~</span><span class="n">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
                                 </span><span class="n">text</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">))})</span><span class="w">


</span><span class="c1"># motifs</span><span class="w">
</span><span class="n">motif</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">as_grob</span><span class="p">(</span><span class="o">~</span><span class="n">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
                 </span><span class="n">text</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"Motif"</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">font</span><span class="o">=</span><span class="m">2</span><span class="p">)))</span><span class="w">

</span><span class="n">motif_logos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">motif_logos</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="n">ggseqlogo</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'prob'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theme_void</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))})</span><span class="w">


</span><span class="c1"># pvalues</span><span class="w">
</span><span class="n">pval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">as_grob</span><span class="p">(</span><span class="o">~</span><span class="n">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
                      </span><span class="n">text</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"p-value"</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">font</span><span class="o">=</span><span class="m">2</span><span class="p">)))</span><span class="w">

</span><span class="n">motif_pval</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">motif_meta</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="n">as_grob</span><span class="p">(</span><span class="o">~</span><span class="n">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">bty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'n'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
                                                             </span><span class="n">text</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">))})</span><span class="w">

</span><span class="c1">######## plot grobs</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'top20_motifs.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">180</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">200</span><span class="p">)</span><span class="w">
</span><span class="n">grid.arrange</span><span class="p">(</span><span class="n">grobs</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">motif_names</span><span class="p">,</span><span class="w"> </span><span class="n">motif</span><span class="p">,</span><span class="w"> </span><span class="n">motif_logos</span><span class="p">,</span><span class="w"> </span><span class="n">pval</span><span class="p">,</span><span class="w"> </span><span class="n">motif_pval</span><span class="p">),</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">widths</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">as.table</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1">########  plot selected motifs</span><span class="w">
</span><span class="n">motifs_of_interest</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'Sox3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Sox2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Sox10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TEAD3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Six2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Six1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Sox9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'AP-2alpha'</span><span class="p">)</span><span class="w">
</span><span class="n">motifs_of_interest</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">motif_meta</span><span class="o">$</span><span class="n">Motif.Name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">motifs_of_interest</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'selected_motifs.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">150</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">150</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">grid.arrange</span><span class="p">(</span><span class="n">grobs</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">motif_names</span><span class="p">[</span><span class="n">motifs_of_interest</span><span class="p">],</span><span class="w"> </span><span class="n">motif</span><span class="p">,</span><span class="w"> </span><span class="n">motif_logos</span><span class="p">[</span><span class="n">motifs_of_interest</span><span class="p">],</span><span class="w"> </span><span class="n">pval</span><span class="p">,</span><span class="w"> </span><span class="n">motif_pval</span><span class="p">[</span><span class="n">motifs_of_interest</span><span class="p">]),</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">widths</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">as.table</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
</details>
</br>
<img class="myImages width_40" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/enhancer_analysis/plot_motifs/output/top20_motifs.png">
<hr />
<br/>
<h2 id="visualising-enhancers-in-the-ucsc-genome-browser-a-nameenhancervisualisationa">Visualising enhancers in the UCSC genome browser <a name="enhancer_visualisation"></a></h2>
</br>
<p>After identifying putative enhancers, we visualised them in the UCSC genome browser before cloning and validating selected enhancers. Below are the genome tracks of validated cloned enhancers and the correponding ChIP/ATAC peaks.</p>
<p><a href="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/enhancer_plots.zip" download>Download all cloned enhancer tracks.</a></p>
<br/>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Lmx1a')">Lmx1a</button>
  <button class="tablinks" onclick="openTab(event, 'Spry2')">Spry2</button>
  <button class="tablinks" onclick="openTab(event, 'Zbtb16')">Zbtb16</button>
  <button class="tablinks" onclick="openTab(event, 'Tfap2a')">Tfap2a</button>
  <button class="tablinks" onclick="openTab(event, 'Sox13')">Sox13</button>
  <button class="tablinks" onclick="openTab(event, 'Epha4')">Epha4</button>
  <button class="tablinks" onclick="openTab(event, 'Pax2')">Pax2</button>
</div>
<br/>
<div id="Lmx1a" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/lmx1a.png">
</div>
<div id="Spry2" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/spry2.png">
</div>
<div id="Zbtb16" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/zbtb16.png">
</div>
<div id="Tfap2a" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/tfap2a.png">
</div>
<div id="Sox13" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/sox13.png">
</div>
<div id="Epha4" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/epha4.png">
</div>
<div id="Pax2" class="tabcontent">
  <img class="myImages width_80" id="myImg" src="https://alexthiery.github.io/otic-reprogramming/assets/enhancer_plots/pax2.png">
</div>
</br>
<!-- The Modal -->
<div id="myModal" class="modal">
  <!-- The Close Button -->
<p><span class="close"></span></p>
  <!-- Modal Content (The Image) -->
  <img class="modal-content" id="img01">
  <!-- Modal Caption (Image Text) -->
  <div id="caption"></div>
</div>

</div>
</div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="https://alexthiery.github.io/otic-reprogramming/public/js/script.js"></script>
  </body>
</html>
