<!DOCTYPE html>
<html lang="en-us">
  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile" />
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />

  <!-- Enable responsiveness on mobile devices-->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, maximum-scale=1"
  />

  <title>
     Smartseq2_downstream &middot; Otic reprogramming 
  </title>

  
  <link rel="canonical" href="http://0.0.0.0:4000/downstream/smartseq2_downstream/" />
  

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css" />
  <link rel="stylesheet" href="/public/css/syntax.css" />
  <link rel="stylesheet" href="/public/css/custom.css" />
  <link rel="stylesheet" href="/public/css/lanyon.css" />
  <link
    rel="stylesheet"
    href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400"
  />

  <!-- stylesheet for icons -->
  <link
    rel="stylesheet"
    href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
  />

  <link rel="shortcut icon" href="/public/favicon.ico" />

  <i class="fas fa-egg"></i>

  <!-- RSS -->
  <link
    rel="alternate"
    type="application/rss+xml"
    title="RSS"
    href="http://0.0.0.0:4000/atom.xml"
  />
</head>


  <body>
    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" />

<!-- Toggleable sidebar -->
<!-- Sidebar icon elements -->

<div class="sidebar" id="sidebar">
  <div class="sidebar-logo">
    <img src="/assets/sidebar_icon.png" />
  </div>

  <div id="contact-list" style="text-align: center">
    
    <a href="https://www.kcl.ac.uk/people/andrea-streit">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-users fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/otic-reprogramming">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-align-center fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="mailto:alexthiery1@gmail.com">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-envelope fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-github-alt fa-stack-1x"></i>
      </span>
    </a>
    
  </div>

  <div class="sidebar-item">
    <p>Documentation for the Buzzi et al. 2021 <a href="https://github.com/alexthiery/otic-reprogramming" target="_blank">Github repository</a>.</p>
  </div>

  <!-- sidebar tabs -->

  <a class="sidebar-nav-item" href="/">Home</a>

  <!-- add tabs automatically from downstream and general collections-->

     
  <a class="sidebar-nav-item" href="/general/quick_start/"
    >Quick start</a
  >
      
  <a class="sidebar-nav-item" href="/general/smartseq2_alignment/"
    >SmartSeq2 alignment</a
  >
      
  <li class="sidebar-nav-item">
    
    <a href="/downstream/downstream_intro/">Downstream analysis</a>
    <ul>
      
      <li class="sidebar-nav-sub-item">
        <a href="/downstream/downstream_intro/">Intro</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="/downstream/lmx1a_downstream/">Lmx1a DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="/downstream/smartseq2_downstream/">SmartSeq2 scRNAseq</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="/downstream/sox8_downstream/">Sox8 DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="/downstream/enhancer_downstream/">Enhancer discovery</a>
      </li>
      
    </ul>
  </li>
  

  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming"
    >GitHub repository</a
  >
  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming/archive/master.zip"
    >Download</a
  >

  <div class="sidebar-item">
    <p>&copy; 2021. All rights reserved.</p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="/" title="Home">Otic reprogramming</a>
            <small>Buzzi et al. 2021</small>
          </h3>
        </div>
      </div>

      <div class="container content"><div class="page">
  <!-- hide auto title generation -->
  <!-- <h1 class="page-title">Smartseq2_downstream</h1> -->
  <h2 id="smartseq2-scrnaseq-analysis">SmartSeq2 scRNAseq analysis</h2>
</br>
<h3 id="load-packages-load-data-and-pre-qc">Load packages, load data and pre-QC</h3>
<details><summary class="box">Expand</summary>
<p>
<p>Automatic switch for running pipeline through Nextflow or interactively in Rstudio</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">getopt</span><span class="p">)</span><span class="w">
</span><span class="n">spec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="w">
  </span><span class="s1">'runtype'</span><span class="p">,</span><span class="w"> </span><span class="s1">'l'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"character"</span><span class="p">,</span><span class="w">
  </span><span class="s1">'cores'</span><span class="w">   </span><span class="p">,</span><span class="w"> </span><span class="s1">'c'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"integer"</span><span class="p">,</span><span class="w">
  </span><span class="s1">'custom_functions'</span><span class="p">,</span><span class="w"> </span><span class="s1">'m'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"character"</span><span class="w">
</span><span class="p">),</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="n">opt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">getopt</span><span class="p">(</span><span class="n">spec</span><span class="p">)</span><span class="w">

</span><span class="c1"># Set run location</span><span class="w">
</span><span class="k">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">){</span><span class="w">
  </span><span class="n">cat</span><span class="p">(</span><span class="s1">'No command line arguments provided, user defaults paths are set for running interactively in Rstudio on docker\n'</span><span class="p">)</span><span class="w">
  </span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"user"</span><span class="w">
</span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"user"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"null"</span><span class="p">){</span><span class="w">
      </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--custom_functions path must be specified in process params config"</span><span class="p">)</span><span class="w">
    </span><span class="p">}</span><span class="w">
  </span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Set paths and pipeline parameters. Load data, packages and custom functions.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"user"</span><span class="p">){</span><span class="w">

    </span><span class="c1"># load custom functions</span><span class="w">
    </span><span class="n">sapply</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s1">'./NF-downstream_analysis/bin/custom_functions/'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">source</span><span class="p">)</span><span class="w">

    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/NF-downstream_analysis/smartseq_analysis/output/"</span><span class="w">
    </span><span class="n">plot_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/NF-downstream_analysis/smartseq_analysis/output/plots/"</span><span class="w">
    </span><span class="n">merged_counts_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/merged_counts/output/'</span><span class="w">
    </span><span class="n">genome_annotations_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-downstream_analysis/extract_gtf_annotations/'</span><span class="w">
    </span><span class="n">gfp_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/merged_counts/output/'</span><span class="w">
    </span><span class="n">velocyto_input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/velocyto/'</span><span class="w">

    </span><span class="c1"># set cores</span><span class="w">
    </span><span class="n">ncores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="w">

  </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">cat</span><span class="p">(</span><span class="s1">'pipeline running through nextflow\n'</span><span class="p">)</span><span class="w">

    </span><span class="c1"># load custom functions</span><span class="w">
    </span><span class="n">sapply</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">source</span><span class="p">)</span><span class="w">
    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/"</span><span class="w">
    </span><span class="n">plot_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/plots/"</span><span class="w">
    </span><span class="n">merged_counts_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">genome_annotations_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">gfp_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">velocyto_input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">

    </span><span class="c1"># set cores</span><span class="w">
    </span><span class="n">ncores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">cores</span><span class="w">
  </span><span class="p">}</span><span class="w">

  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

  </span><span class="c1"># load required packages</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">Antler</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">velocyto.R</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">monocle</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">plyr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">ggsignif</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">rstatix</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># set pipeline params</span><span class="w">
</span><span class="n">seed</span><span class="o">=</span><span class="m">1</span><span class="w">
</span><span class="n">perp</span><span class="o">=</span><span class="m">5</span><span class="w">
</span><span class="n">eta</span><span class="o">=</span><span class="m">200</span><span class="w">

</span><span class="cd">#' Stage colors</span><span class="w">
</span><span class="n">stage_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"#BBBDC1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#6B98E9"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#05080D"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'11'</span><span class="p">,</span><span class="w"> </span><span class="s1">'15'</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Load data and initialise Antler object</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Load and hygienize dataset</span><span class="w">
</span><span class="n">m</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Antler</span><span class="o">$</span><span class="n">new</span><span class="p">(</span><span class="n">plot_folder</span><span class="o">=</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="n">num_cores</span><span class="o">=</span><span class="n">ncores</span><span class="p">)</span><span class="w">

</span><span class="c1"># load in phenoData and assayData from ../dataset -&gt; assayData is count matrix; phenoData is metaData (i.e. replicated, conditions, samples etc)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">loadDataset</span><span class="p">(</span><span class="n">folderpath</span><span class="o">=</span><span class="n">merged_counts_path</span><span class="p">)</span><span class="w">

</span><span class="n">pData</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">cells_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">[</span><span class="nf">as.character</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">)]</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot pre-QC metrics</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">plotReadcountStats</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Raw"</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">category</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s2">"preQC"</span><span class="p">,</span><span class="w"> </span><span class="n">reads_name</span><span class="o">=</span><span class="s2">"read"</span><span class="p">,</span><span class="w"> </span><span class="n">cat_colors</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Cells per stage-pre')">Cells per stage</button>
  <button class="tablinks" onclick="openTab(event, 'Readcounts per cell-pre')">Readcounts per cell</button>
  <button class="tablinks" onclick="openTab(event, 'Genes per cell-pre')">Genes per cell</button>
</div>
</br>
<div id="Cells per stage-pre" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_cellNumber_timepoint_by_timepoint.png">
</div>
<div id="Readcounts per cell-pre" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_counts_timepoint_by_timepoint.png">
</div>
<div id="Genes per cell-pre" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_geneCounts_timepoint_by_timepoint.png">
</div>
</br>
</br>
<p>Some key genes are not annotated in the GTF - manually add these gene names to the annotations file</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># read in annotations file</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">genome_annotations_path</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'*gene_annotations.csv'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">

</span><span class="c1"># add missing annotations to annotations file</span><span class="w">
</span><span class="n">extra_annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'FOXI3'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000037457'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ATN1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000014554'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX10'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000038767'</span><span class="p">,</span><span class="w">
                      </span><span class="s1">'COL11A1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000005180'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GRHL2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000037687'</span><span class="p">)</span><span class="w">

</span><span class="c1"># add extra annotations to annotations csv file</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">gtf_annotations</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">extra_annotations</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">extra_annotations</span><span class="p">)[</span><span class="n">extra_annotations</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">x</span><span class="p">],</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">2</span><span class="p">]))</span><span class="w">

</span><span class="c1"># label extra MT genes</span><span class="w">
</span><span class="n">MT_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'ND3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CYTB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'COII'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ATP8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ND4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ND4L'</span><span class="p">)</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="p">[</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">MT_genes</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s1">'MT-'</span><span class="p">,</span><span class="w"> </span><span class="n">gtf_annotations</span><span class="p">[</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">MT_genes</span><span class="p">,</span><span class="m">2</span><span class="p">])</span><span class="w">

</span><span class="n">write.csv</span><span class="p">(</span><span class="n">gtf_annotations</span><span class="p">,</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'new_annotations.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">

</span><span class="c1"># set gene annotations</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">setCurrentGeneNames</span><span class="p">(</span><span class="n">geneID_mapping_file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'new_annotations.csv'</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Add list of key genes to Antler object</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cd">#' Store known genes</span><span class="w">
</span><span class="n">apriori_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="w">
  </span><span class="s1">'DACH1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXI3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GATA3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GBX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HESX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SALL4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOHO-1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Known_otic_genes</span><span class="w">
  </span><span class="s1">'ATN1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'BACH2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CNOT4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CXCL14'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DACH2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FEZ1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXP3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HIPK1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HOMER2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'KLF7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'KREMEN1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LDB1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MSI1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PDLIM4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PLAG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PNOC'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PKNOX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'RERE'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SMOC1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX13'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TCF7L2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TEAD3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZBTB16'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZFHX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZNF384'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZNF385C'</span><span class="p">,</span><span class="w"> </span><span class="c1"># New_otic_TFs</span><span class="w">
  </span><span class="s1">'PRDM1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXI1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NKX2-6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NR2F2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PDLIM1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PHOX2B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SALL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2E'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TLX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'VGLL2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Epibranchial_genes</span><span class="w">
  </span><span class="s1">'ZNF423'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CXCR4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAFB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYC'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZEB2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Neural_Genes</span><span class="w">
  </span><span class="s1">'CD151'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETS2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FGFR4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'OTX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Pax3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX3'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Non_otic_placode_genes</span><span class="w">
  </span><span class="s1">'FOXD3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ID2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ID4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MSX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2C'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Neural_Crest_Genes</span><span class="w">
  </span><span class="s1">'GATA2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Epidermis_genes</span><span class="w">
  </span><span class="s1">'COL11A1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DTX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GRHL2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NELL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'OTOL1'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Disease_associated_genes</span><span class="w">
  </span><span class="s1">'ARID3A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'BMP4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CREBBP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETV4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETV5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXP4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FSTL4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HOXA2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'JAG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LFNG'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LZTS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAFA'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MEIS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYCN'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NFKB1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NOTCH1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SPRY1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SPRY2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SSTR5'</span><span class="p">,</span><span class="w"> </span><span class="c1"># chen et al. 2017</span><span class="w">
  </span><span class="s1">'TWIST1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ASL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAF'</span><span class="w"> </span><span class="c1"># other_genes</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">favorite_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">sort</span><span class="p">(</span><span class="n">apriori_genes</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</details>
<hr />
</br>
<h3 id="data-pre-processing-and-qc">Data pre-processing and QC</h3>
</br>
<details><summary class="box">Expand</summary>
<p>
<p>Remove gene and cell outliers</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cd">#' Remove outliers genes and cells</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">removeOutliers</span><span class="p">(</span><span class="w"> </span><span class="n">lowread_thres</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5e5</span><span class="p">,</span><span class="w">   </span><span class="c1"># select cells with more than 500000 reads</span><span class="w">
                  </span><span class="n">genesmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span><span class="w">       </span><span class="c1"># select cells expressing more than 1k genes</span><span class="w">
                  </span><span class="n">cellmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w">           </span><span class="c1"># select genes expressed in more than 3 cells)</span><span class="w">
                  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Raw'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Remove control cells</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
  </span><span class="s2">"blank"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'241112'</span><span class="p">,</span><span class="w"> </span><span class="s1">'250184'</span><span class="p">,</span><span class="w"> </span><span class="s1">'265102'</span><span class="p">,</span><span class="w"> </span><span class="s1">'272111'</span><span class="p">,</span><span class="w"> </span><span class="s1">'248185'</span><span class="p">,</span><span class="w"> </span><span class="s1">'274173'</span><span class="p">),</span><span class="w">
  </span><span class="s2">"bulk"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'225110'</span><span class="p">,</span><span class="w"> </span><span class="s1">'251172'</span><span class="p">,</span><span class="w"> </span><span class="s1">'273103'</span><span class="p">,</span><span class="w"> </span><span class="s1">'280110'</span><span class="p">,</span><span class="w"> </span><span class="s1">'235161'</span><span class="p">,</span><span class="w"> </span><span class="s1">'246161'</span><span class="p">),</span><span class="w">
  </span><span class="s2">"human"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'233111'</span><span class="p">,</span><span class="w"> </span><span class="s1">'249196'</span><span class="p">,</span><span class="w"> </span><span class="s1">'257101'</span><span class="p">,</span><span class="w"> </span><span class="s1">'264112'</span><span class="p">,</span><span class="w"> </span><span class="s1">'233185'</span><span class="p">,</span><span class="w"> </span><span class="s1">'247173'</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">excludeCellsFromIds</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">getCellsNames</span><span class="p">()</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">annotations</span><span class="p">)))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Remove cells with more than 6% of mitochondrial read counts</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">removeGenesFromRatio</span><span class="p">(</span><span class="w">
  </span><span class="n">candidate_genes</span><span class="o">=</span><span class="n">grep</span><span class="p">(</span><span class="s1">'^MT-'</span><span class="p">,</span><span class="w"> </span><span class="n">m</span><span class="o">$</span><span class="n">getGeneNames</span><span class="p">(),</span><span class="w"> </span><span class="n">value</span><span class="o">=</span><span class="nb">T</span><span class="p">),</span><span class="w">
  </span><span class="n">threshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.06</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>QC plots</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">plotReadcountStats</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Raw"</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">category</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s2">"postQC"</span><span class="p">,</span><span class="w"> </span><span class="n">reads_name</span><span class="o">=</span><span class="s2">"read"</span><span class="p">,</span><span class="w"> </span><span class="n">cat_colors</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Cells per stage-post')">Cells per stage</button>
  <button class="tablinks" onclick="openTab(event, 'Readcounts per cell-post')">Readcounts per cell</button>
  <button class="tablinks" onclick="openTab(event, 'Genes per cell-post')">Genes per cell</button>
</div>
</br>
<div id="Cells per stage-post" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/postQC_statistics_cellNumber_timepoint_by_timepoint.png">
</div>
<div id="Readcounts per cell-post" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/postQC_statistics_counts_timepoint_by_timepoint.png">
</div>
<div id="Genes per cell-post" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/postQC_statistics_geneCounts_timepoint_by_timepoint.png">
</div>
</br>
</br>
<p>Normalize readcounts to CPM and remove genes with &lt;10 CPM</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">normalize</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s2">"Count-Per-Million"</span><span class="p">)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">excludeUnexpressedGenes</span><span class="p">(</span><span class="n">min.cells</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">removeLowlyExpressedGenes</span><span class="p">(</span><span class="n">expression_threshold</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">selection_theshold</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</details>
<hr />
</br>
<h3 id="transcriptomic-analysis-of-all-cells">Transcriptomic analysis of all cells</h3>
</br>
<details><summary class="box">Expand</summary>
<p>
<p>Generate gene-gene correlation matrix</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># change plot folder</span><span class="w">
</span><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"all_cells/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="c1"># "fastCor" uses the tcrossprod function which by default relies on BLAS to speed up computation. This produces inconsistent result in precision depending on the version of BLAS being used.</span><span class="w">
</span><span class="c1"># see "matprod" in https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html</span><span class="w">
</span><span class="n">corr.mat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fastCor</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)),</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s2">"spearman"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p><strong><em>Identification of modules of co-correlated genes</em></strong></p>
<p>This feature is the prime reason for using the Antler package. Using the Antler package, we are able to cluster genes and identify sets of co-correlated genes termed <em>gene modules</em>. This works by heirarchically clustering a gene-gene correlation matrix, followed by iterative filtering of poor quality clusters.</p>
<p>For further information on Antler gene module identification, click <a href="https://juliendelile.github.io/Antler/articles/Transcriptomic-summary.html#gene-modules-identification">here</a>.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cd">#' ## Gene modules identification</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">identifyGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">method</span><span class="o">=</span><span class="s2">"TopCorr_DR"</span><span class="p">,</span><span class="w">
  </span><span class="n">corr</span><span class="o">=</span><span class="n">corr.mat</span><span class="p">,</span><span class="w">
  </span><span class="n">corr_t</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">,</span><span class="w"> </span><span class="c1"># gene correlation threshold</span><span class="w">
  </span><span class="n">topcorr_mod_min_cell</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_mod_consistency_thres</span><span class="o">=</span><span class="m">0.4</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_mod_skewness_thres</span><span class="o">=-</span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_min_cell_level</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w">
  </span><span class="n">topcorr_num_max_final_gms</span><span class="o">=</span><span class="m">40</span><span class="p">,</span><span class="w"> </span><span class="c1"># heuristically set to give sufficient cluster diversity and of reasonable size in order to select modules from gene candidate list</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="c1"># corr_t = gene correlation threshold</span><span class="w">
</span><span class="c1"># topcorr_mod_min_cell = minimum number of cells expressing gene module</span><span class="w">
</span><span class="c1"># topcorr_mod_consistency_thres = proportion of cells expressing gene module (binarised z-scored log-transformed normalised expression levels)</span><span class="w">
</span><span class="c1"># topcorr_mod_skewness_thres = test whether genes are expressed only in subset of cells</span><span class="w">
</span><span class="c1"># topcorr_min_cell_level = minimum expression level in cells which express gene module</span><span class="w">
</span><span class="c1"># topcorr_num_max_final_gms = maximum number of final gene modules</span><span class="w">

</span><span class="nf">names</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"GM "</span><span class="p">,</span><span class="w"> </span><span class="n">seq</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot gene modules</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># identify cell clusters based on remaining genes</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">identifyCellClusters</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w"> </span><span class="n">used_genes</span><span class="o">=</span><span class="s2">"topCorr_DR.genemodules"</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">plotGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">basename</span><span class="o">=</span><span class="s1">'AllCells'</span><span class="p">,</span><span class="w">
  </span><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.gms</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'topCorr_DR.genemodules'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.geneset</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w">
  </span><span class="n">use.dendrogram</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w">
  </span><span class="n">display.clusters</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">file_settings</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">type</span><span class="o">=</span><span class="s1">'pdf'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">20</span><span class="p">)),</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w">
  </span><span class="n">gene_transformations</span><span class="o">=</span><span class="s1">'logscaled'</span><span class="p">,</span><span class="w">
  </span><span class="n">pretty.params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"size_factor"</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"ngenes_per_lines"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s2">"side.height.fraction"</span><span class="o">=</span><span class="m">.15</span><span class="p">),</span><span class="w">
  </span><span class="n">display.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">extra_legend</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"text"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'ss8-9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss11-12'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss14-15'</span><span class="p">),</span><span class="w"> </span><span class="s2">"colors"</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">writeGeneModules</span><span class="p">(</span><span class="n">basename</span><span class="o">=</span><span class="s1">'AllCells_allGms'</span><span class="p">,</span><span class="w"> </span><span class="n">gms</span><span class="o">=</span><span class="s1">'topCorr_DR.genemodules'</span><span class="p">,</span><span class="w"> </span><span class="n">folder_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCells_topCorr_DR.genemodules_Normalized_logscaled.pdf" download>Download
PDF</a></p>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCells_allGms_topCorr_DR.genemodules.txt" download>Download
gene module list</a></p>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCells_topCorr_DR.genemodules_Normalized_logscaled.png">
</br>
</br>
<p>Filter out cells with low summary counts</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">m</span><span class="o">$</span><span class="n">copy</span><span class="p">()</span><span class="w">
</span><span class="n">m2</span><span class="o">$</span><span class="n">excludeCellsFromIds</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">unlist</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">),]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">colSums</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="p">{</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">scale</span><span class="p">(</span><span class="nf">log</span><span class="p">(</span><span class="n">.</span><span class="p">),</span><span class="w"> </span><span class="n">center</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">scale</span><span class="o">=</span><span class="nb">T</span><span class="p">))}</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="p">{</span><span class="n">which</span><span class="p">(</span><span class="n">.</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">-1.5</span><span class="p">)})</span><span class="w">
</span><span class="n">m2</span><span class="o">$</span><span class="n">excludeUnexpressedGenes</span><span class="p">(</span><span class="n">min.cells</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Normalized"</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Select gene modules based on the presence of genes known to be involved in differentiation and re-cluster cells</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">bait_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"HOXA2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PAX6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MSX1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Pax3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SALL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ETS1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TWIST1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"HOMER2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">,</span><span class="w"> </span><span class="s2">"VGLL2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"EYA2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PRDM1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NELL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"DLX5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX10"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOHO-1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"IRX4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"DLX6"</span><span class="p">)</span><span class="w">

</span><span class="n">m2</span><span class="o">$</span><span class="n">dR</span><span class="o">$</span><span class="n">genemodules</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Filter</span><span class="p">(</span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">any</span><span class="p">(</span><span class="n">bait_genes</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">x</span><span class="p">)},</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)</span><span class="w">

</span><span class="c1"># cluster into 5 clusters</span><span class="w">
</span><span class="n">m2</span><span class="o">$</span><span class="n">identifyCellClusters</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w"> </span><span class="n">clust_name</span><span class="o">=</span><span class="s2">"Mansel"</span><span class="p">,</span><span class="w"> </span><span class="n">used_genes</span><span class="o">=</span><span class="s2">"dR.genemodules"</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w"> </span><span class="n">numclusters</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">

</span><span class="c1"># set cluster colours</span><span class="w">
</span><span class="n">clust.colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'#da70d6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#c71585'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#b0c4de'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#afeeee'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#5f9ea0'</span><span class="p">)</span><span class="w">

</span><span class="c1"># read in GFP counts</span><span class="w">
</span><span class="n">gfp_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">gfp_counts</span><span class="p">,</span><span class="w"> </span><span class="s1">'gfpData.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">check.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot final clustering of all cells</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m2</span><span class="o">$</span><span class="n">plotGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">basename</span><span class="o">=</span><span class="s1">'AllCellsManualGMselection'</span><span class="p">,</span><span class="w">
  </span><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.gms</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'dR.genemodules'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.geneset</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w">
  </span><span class="n">use.dendrogram</span><span class="o">=</span><span class="s1">'Mansel'</span><span class="p">,</span><span class="w">
  </span><span class="n">file_settings</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">type</span><span class="o">=</span><span class="s1">'pdf'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">)),</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w">
  </span><span class="n">gene_transformations</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'log'</span><span class="p">,</span><span class="w"> </span><span class="s1">'logscaled'</span><span class="p">),</span><span class="w">
  </span><span class="n">extra_colors</span><span class="o">=</span><span class="n">cbind</span><span class="p">(</span><span class="w">
    </span><span class="n">m2</span><span class="o">$</span><span class="n">cellClusters</span><span class="o">$</span><span class="n">Mansel</span><span class="o">$</span><span class="n">cell_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">clust.colors</span><span class="p">[</span><span class="n">.</span><span class="p">],</span><span class="w">
</span><span class="s2">"PAX2_log"</span><span class="o">=</span><span class="n">m2</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="s1">'PAX2'</span><span class="p">,]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="p">{</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">.</span><span class="p">)}</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="p">{</span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">.</span><span class="o">/</span><span class="nf">max</span><span class="p">(</span><span class="n">.</span><span class="p">))}</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="s2">"black"</span><span class="p">))(</span><span class="n">n</span><span class="o">=</span><span class="m">100</span><span class="p">)[</span><span class="n">.</span><span class="p">],</span><span class="w">
    </span><span class="s2">"GFP_log"</span><span class="o">=</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">gfp_counts</span><span class="p">[</span><span class="n">m2</span><span class="o">$</span><span class="n">getCellsNames</span><span class="p">()])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
</span><span class="p">{</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">.</span><span class="p">)}</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
</span><span class="p">{</span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">.</span><span class="o">/</span><span class="nf">max</span><span class="p">(</span><span class="n">.</span><span class="p">))}</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
</span><span class="n">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="s2">"darkgreen"</span><span class="p">))(</span><span class="n">n</span><span class="o">=</span><span class="m">100</span><span class="p">)[</span><span class="n">.</span><span class="p">]</span><span class="w">
</span><span class="p">),</span><span class="w">
</span><span class="n">pretty.params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"size_factor"</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"side.height.fraction"</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
</span><span class="n">display.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
</span><span class="n">extra_legend</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"text"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'ss8-9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss11-12'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss14-15'</span><span class="p">),</span><span class="w"> </span><span class="s2">"colors"</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m2</span><span class="o">$</span><span class="n">writeGeneModules</span><span class="p">(</span><span class="n">basename</span><span class="o">=</span><span class="s1">'AllCells_baitGMs'</span><span class="p">,</span><span class="w"> </span><span class="n">gms</span><span class="o">=</span><span class="s1">'dR.genemodules'</span><span class="p">,</span><span class="w"> </span><span class="n">folder_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCellsManualGMselection_dR.genemodules_Normalized_logscaled.pdf" download>Download
PDF</a></p>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCells_baitGMs_dR.genemodules" download>Download
gene module list</a></p>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/AllCellsManualGMselection_dR.genemodules_Normalized_logscaled.png">
</br>
</br>
<p>Plot tSNEs of cell clusters and developmental stage</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"all_cells/tsne/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">


</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'allcells_clusters_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m2</span><span class="p">,</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">dR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">m2</span><span class="o">$</span><span class="n">cellClusters</span><span class="o">$</span><span class="n">Mansel</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="o">=</span><span class="n">clust.colors</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">


</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'allcells_stage_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m2</span><span class="p">,</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">dR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pData</span><span class="p">(</span><span class="n">m2</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Developmental stage'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'allcells_clusters_TSNE')">Cell clusters tSNE</button>
  <button class="tablinks" onclick="openTab(event, 'allcells_stage_TSNE')">Developmental stage tSNE</button>
</div>
</br>
<div id="allcells_clusters_TSNE" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/allcells_clusters_TSNE.png">
</div>
<div id="allcells_stage_TSNE" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/allcells_stage_TSNE.png">
</div>
</br>
</br>
<p>Plot tSNEs of genes of interest</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">gene_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'SOX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX21'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX1'</span><span class="p">)</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">gn</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">gene_list</span><span class="p">){</span><span class="w">
  </span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">gn</span><span class="p">,</span><span class="w"> </span><span class="s1">'_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m2</span><span class="p">,</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">dR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m2</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,])</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m2</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,]))),</span><span class="w">
            </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> </span><span class="s2">"darkmagenta"</span><span class="p">),</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">ggtitle</span><span class="p">(</span><span class="n">gn</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)))</span><span class="w">
  </span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span><span class="p">}</span><span class="w">


</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'GFP_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m2</span><span class="p">,</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">dR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">gfp_counts</span><span class="p">[</span><span class="n">m2</span><span class="o">$</span><span class="n">getCellsNames</span><span class="p">()])</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">gfp_counts</span><span class="p">[</span><span class="n">m2</span><span class="o">$</span><span class="n">getCellsNames</span><span class="p">()]))),</span><span class="w">
                </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> </span><span class="s2">"darkgreen"</span><span class="p">),</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'GFP'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'all_SOX2')">Sox2</button>
  <button class="tablinks" onclick="openTab(event, 'all_SOX10')">Sox10</button>
  <button class="tablinks" onclick="openTab(event, 'all_SOX8')">Sox8</button>
  <button class="tablinks" onclick="openTab(event, 'all_PAX7')">Pax7</button>
  <button class="tablinks" onclick="openTab(event, 'all_PAX2')">Pax2</button>
  <button class="tablinks" onclick="openTab(event, 'all_LMX1A')">Lmx1a</button>
  <button class="tablinks" onclick="openTab(event, 'all_SOX21')">Sox21</button>
  <button class="tablinks" onclick="openTab(event, 'all_SIX1')">Six1</button>
  <button class="tablinks" onclick="openTab(event, 'all_GFP')">GFP</button>
</div>
</br>
<div id="all_SOX2" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/SOX2_TSNE.png">
</div>
<div id="all_SOX10" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/SOX10_TSNE.png">
</div>
<div id="all_SOX8" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/SOX8_TSNE.png">
</div>
<div id="all_PAX7" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/PAX7_TSNE.png">
</div>
<div id="all_PAX2" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/PAX2_TSNE.png">
</div>
<div id="all_LMX1A" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/LMX1A_TSNE.png">
</div>
<div id="all_SOX21" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/SOX21_TSNE.png">
</div>
<div id="all_SIX1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/SIX1_TSNE.png">
</div>
<div id="all_GFP" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/tsne/GFP_TSNE.png">
</div>
</br>
</br>
<p>Plot dotplot for genes of interest</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"all_cells/"</span><span class="p">)</span><span class="w">
</span><span class="c1"># gene list for dotplot</span><span class="w">
</span><span class="n">gene_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"DLX6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"HOMER2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TFAP2E"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ZNF385C"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SIX1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PAX2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"DLX3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NELL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"FGF8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"VGLL2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"EYA1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOHO-1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ZBTB16"</span><span class="p">,</span><span class="w"> </span><span class="s2">"DLX5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TFAP2A"</span><span class="p">,</span><span class="w"> </span><span class="c1"># Placodes</span><span class="w">
              </span><span class="s2">"SOX10"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WNT1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MSX2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"BMP5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PAX7"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TFAP2B"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMO4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ETS1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MSX1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX9"</span><span class="p">,</span><span class="w"> </span><span class="c1"># NC</span><span class="w">
              </span><span class="s2">"ZEB2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"HOXA2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"RFX4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PAX6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WNT4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX21"</span><span class="p">,</span><span class="w"> </span><span class="c1"># Neural</span><span class="w">
              </span><span class="s2">"SIM1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PITX2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TWIST1"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Mesoderm</span><span class="w">

</span><span class="c1"># get cell branch information for dotplot</span><span class="w">
</span><span class="n">cell_cluster_data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">cellClusters</span><span class="o">$</span><span class="n">Mansel</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tibble</span><span class="o">::</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="s1">'cellname'</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">celltype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">case_when</span><span class="p">(</span><span class="w">
    </span><span class="n">cluster</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"1"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"OEP"</span><span class="p">,</span><span class="w">
    </span><span class="n">cluster</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"2"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"Late Placodal"</span><span class="p">,</span><span class="w">
    </span><span class="n">cluster</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"3"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s1">'Neural'</span><span class="p">,</span><span class="w">
    </span><span class="n">cluster</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"4"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s1">'Mesodermal'</span><span class="p">,</span><span class="w">
    </span><span class="n">cluster</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"5"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s1">'Neural Crest'</span><span class="w">
  </span><span class="p">))</span><span class="w">

</span><span class="c1"># gather data for dotplot</span><span class="w">
</span><span class="n">dotplot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">m2</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gene_list</span><span class="p">,</span><span class="w"> </span><span class="p">]),</span><span class="w"> </span><span class="n">check.names</span><span class="o">=</span><span class="nb">F</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tibble</span><span class="o">::</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="s1">'cellname'</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tidyr</span><span class="o">::</span><span class="n">gather</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">cellname</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">left_join</span><span class="p">(</span><span class="n">cell_cluster_data</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"cellname"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># calculate percentage of cells in each cluster expressing gene</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="s1">'Proportion of Cells Expressing'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="o">/</span><span class="n">n</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># scale data</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">scale</span><span class="p">(</span><span class="n">value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># calculate mean expression</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="s1">'Scaled Average Expression'</span><span class="o">=</span><span class="n">mean</span><span class="p">(</span><span class="n">value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">distinct</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># make factor levels to order genes in dotplot</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">genename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_list</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># make factor levels to order cells in dotplott</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">celltype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rev</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Late Placodal"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Neural Crest"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Neural"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Mesodermal"</span><span class="p">))))</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s2">"all_cells_dotplot.png"</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">25</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">dotplot_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="n">`Proportion of Cells Expressing`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">`Scaled Average Expression`</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_count</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_size_area</span><span class="p">(</span><span class="n">max_size</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_x_discrete</span><span class="p">(</span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlab</span><span class="p">(</span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ylab</span><span class="p">(</span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey90"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">9</span><span class="p">),</span><span class="w"> </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clust.colors</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">5</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">)],</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'bold'</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">),</span><span class="w">
        </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"bottom"</span><span class="p">,</span><span class="w"> </span><span class="n">legend.box</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"horizontal"</span><span class="p">,</span><span class="w"> </span><span class="n">plot.margin</span><span class="o">=</span><span class="n">unit</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<img class="myImages width_90" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/all_cells/all_cells_dotplot.png">
</details>
<hr />
</br>
<h3 id="transcriptomic-analysis-of-oep-cells">Transcriptomic analysis of OEP cells</h3>
<details><summary class="box">Expand</summary>
<p>
<p>Clusters 3-5 are composed of non-oep derived populations (they are also mostly PAX2 negative)</p>
<p>We exclude these cells from the subsequent analysis</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"oep_subset/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">m2</span><span class="o">$</span><span class="n">copy</span><span class="p">()</span><span class="w">
</span><span class="n">m_oep</span><span class="o">$</span><span class="n">excludeCellFromClusterIds</span><span class="p">(</span><span class="n">cluster_ids</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">3</span><span class="o">:</span><span class="m">5</span><span class="p">),</span><span class="w"> </span><span class="n">used_clusters</span><span class="o">=</span><span class="s1">'Mansel'</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">

</span><span class="cd">#' Some genes may not be expressed any more in the remaining cells</span><span class="w">
</span><span class="n">m_oep</span><span class="o">$</span><span class="n">excludeUnexpressedGenes</span><span class="p">(</span><span class="n">min.cells</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Normalized"</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">m_oep</span><span class="o">$</span><span class="n">removeLowlyExpressedGenes</span><span class="p">(</span><span class="n">expression_threshold</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">selection_theshold</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Calculate and plot gene modules for OEPs</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">
</span><span class="c1"># "fastCor" uses the tcrossprod function which by default relies on BLAS to speed up computation. This produces inconsistent result in precision depending on the version of BLAS being used.</span><span class="w">
</span><span class="c1"># see "matprod" in https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html</span><span class="w">
</span><span class="n">corr.mat2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fastCor</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)),</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s2">"spearman"</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">identifyGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">method</span><span class="o">=</span><span class="s2">"TopCorr_DR"</span><span class="p">,</span><span class="w">
  </span><span class="n">corr</span><span class="o">=</span><span class="n">corr.mat2</span><span class="p">,</span><span class="w">
  </span><span class="n">corr_t</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">,</span><span class="w">
  </span><span class="n">topcorr_mod_min_cell</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_mod_consistency_thres</span><span class="o">=</span><span class="m">0.4</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_mod_skewness_thres</span><span class="o">=-</span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="c1"># default</span><span class="w">
  </span><span class="n">topcorr_min_cell_level</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="nf">names</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"GM "</span><span class="p">,</span><span class="w"> </span><span class="n">seq</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)))</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">writeGeneModules</span><span class="p">(</span><span class="n">folder_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s1">'OEP_allGms'</span><span class="p">,</span><span class="w"> </span><span class="n">gms</span><span class="o">=</span><span class="s1">'topCorr_DR.genemodules'</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">identifyCellClusters</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w"> </span><span class="n">used_genes</span><span class="o">=</span><span class="s2">"topCorr_DR.genemodules"</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">plotGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w">
  </span><span class="n">basename</span><span class="o">=</span><span class="s1">'OEP_allGms'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.gms</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'topCorr_DR.genemodules'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.geneset</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w">
  </span><span class="n">use.dendrogram</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w">
  </span><span class="n">display.clusters</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">file_settings</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">type</span><span class="o">=</span><span class="s1">'pdf'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">)),</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w">
  </span><span class="n">gene_transformations</span><span class="o">=</span><span class="s1">'logscaled'</span><span class="p">,</span><span class="w">
  </span><span class="n">pretty.params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"size_factor"</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"ngenes_per_lines"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s2">"side.height.fraction"</span><span class="o">=</span><span class="m">0.15</span><span class="p">),</span><span class="w">
  </span><span class="n">display.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">extra_legend</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"text"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'ss8-9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss11-12'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss14-15'</span><span class="p">),</span><span class="w"> </span><span class="s2">"colors"</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_allGms_topCorr_DR.genemodules_Normalized_logscaled.pdf" download>Download
PDF</a></p>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_allGms_topCorr_DR.genemodules.txt" download>Download
gene module list</a></p>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_allGms_topCorr_DR.genemodules_Normalized_logscaled.png">
</br>
</br>
<p>Select gene modules based on the presence of genes known to be involved in otic and epibranchial differentiation and re-cluster cells</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">bait_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"HOMER2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOHO-1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX10"</span><span class="p">,</span><span class="w"> </span><span class="s2">"VGLL2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZNF385C'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NELL1'</span><span class="p">,</span><span class="w"> </span><span class="s2">"CXCL14"</span><span class="p">,</span><span class="w"> </span><span class="s2">"EYA4"</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Filter</span><span class="p">(</span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">any</span><span class="p">(</span><span class="n">bait_genes</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">x</span><span class="p">)},</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules</span><span class="p">)</span><span class="w">

</span><span class="c1"># cluster into 5 clusters</span><span class="w">
</span><span class="n">m_oep</span><span class="o">$</span><span class="n">identifyCellClusters</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s1">'hclust'</span><span class="p">,</span><span class="w"> </span><span class="n">clust_name</span><span class="o">=</span><span class="s2">"Mansel"</span><span class="p">,</span><span class="w"> </span><span class="n">used_genes</span><span class="o">=</span><span class="s2">"topCorr_DR.genemodules.selected"</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w"> </span><span class="n">numclusters</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">

</span><span class="n">clust.colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'#ffa07a'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#f55f20'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#dda0dd'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#48d1cc'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#b2ffe5'</span><span class="p">)</span><span class="w">

</span><span class="n">m_oep</span><span class="o">$</span><span class="n">plotGeneModules</span><span class="p">(</span><span class="w">
  </span><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w">
  </span><span class="n">basename</span><span class="o">=</span><span class="s1">'OEP_GMselection'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.gms</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'topCorr_DR.genemodules.selected'</span><span class="p">,</span><span class="w">
  </span><span class="n">displayed.geneset</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w">
  </span><span class="n">use.dendrogram</span><span class="o">=</span><span class="s1">'Mansel'</span><span class="p">,</span><span class="w">
  </span><span class="n">file_settings</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">type</span><span class="o">=</span><span class="s1">'pdf'</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">5</span><span class="p">)),</span><span class="w">
  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">,</span><span class="w">
  </span><span class="n">gene_transformations</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'log'</span><span class="p">,</span><span class="w"> </span><span class="s1">'logscaled'</span><span class="p">),</span><span class="w">
  </span><span class="n">extra_colors</span><span class="o">=</span><span class="n">cbind</span><span class="p">(</span><span class="w">
    </span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="o">$</span><span class="n">Mansel</span><span class="o">$</span><span class="n">cell_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">clust.colors</span><span class="p">[</span><span class="n">.</span><span class="p">]</span><span class="w">
  </span><span class="p">),</span><span class="w">
  </span><span class="n">display.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">pretty.params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"size_factor"</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"ngenes_per_lines"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="s2">"side.height.fraction"</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span><span class="w">
  </span><span class="n">extra_legend</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="s2">"text"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'ss8-9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss11-12'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ss14-15'</span><span class="p">),</span><span class="w"> </span><span class="s2">"colors"</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># add gene modules txt</span><span class="w">
</span><span class="n">m_oep</span><span class="o">$</span><span class="n">writeGeneModules</span><span class="p">(</span><span class="n">folder_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s1">'OEP_GMselection'</span><span class="p">,</span><span class="w"> </span><span class="n">gms</span><span class="o">=</span><span class="s1">'topCorr_DR.genemodules'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_GMselection_topCorr_DR.genemodules.selected_Normalized_logscaled.pdf" download>Download
PDF</a></p>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_GMselection_topCorr_DR.genemodules.txt" download>Download
gene module list</a></p>
<img class="myImages width_90" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/OEP_GMselection_topCorr_DR.genemodules.selected_Normalized_logscaled.png">
</br>
</br>
<p>Plot tSNEs of cell clusters and developmental stage</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'oep_subset/tsne/'</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_clusters_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="p">[[</span><span class="s1">'Mansel'</span><span class="p">]]</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="o">=</span><span class="n">clust.colors</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_stage_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Developmental stage'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'oep_clusters_TSNE')">Cell clusters tSNE</button>
  <button class="tablinks" onclick="openTab(event, 'oep_stage_TSNE')">Developmental stage tSNE</button>
</div>
</br>
<div id="oep_clusters_TSNE" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/OEP_clusters_TSNE.png">
</div>
<div id="oep_stage_TSNE" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/OEP_stage_TSNE.png">
</div>
</br>
</br>
<p>Plot expression of bait genes plus genes from bulk RNAseq and literature on tsne</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">gene_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">bait_genes</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2E'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZBTB16'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PDLIM1'</span><span class="p">)</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">gn</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">gene_list</span><span class="p">){</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">gn</span><span class="p">,</span><span class="w"> </span><span class="s1">'\_TSNE.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,])</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,]))),</span><span class="w">
</span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> </span><span class="s2">"darkmagenta"</span><span class="p">),</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">ggtitle</span><span class="p">(</span><span class="n">gn</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/oep_GOI_tSNEs.zip" download>Download
all tSNEs</a></p>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'oep_SOX8')">Sox2</button>
  <button class="tablinks" onclick="openTab(event, 'oep_PAX2')">Pax2</button>
  <button class="tablinks" onclick="openTab(event, 'oep_TFAP2E')">TFAP2E</button>
  <button class="tablinks" onclick="openTab(event, 'oep_SIX1')">Six1</button>
  <button class="tablinks" onclick="openTab(event, 'oep_ZBTB16')">ZBTB16</button>
  <button class="tablinks" onclick="openTab(event, 'oep_FOXG1')">FOXG1</button>
  <button class="tablinks" onclick="openTab(event, 'oep_PDLIM1')">PDLIM1</button>
</div>
</br>
<div id="oep_SOX8" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/SOX8_TSNE.png">
</div>
<div id="oep_PAX2" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/PAX2_TSNE.png">
</div>
<div id="oep_TFAP2E" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/TFAP2E_TSNE.png">
</div>
<div id="oep_SIX1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/SIX1_TSNE.png">
</div>
<div id="oep_ZBTB16" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/ZBTB16_TSNE.png">
</div>
<div id="oep_FOXG1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/FOXG1_TSNE.png">
</div>
<div id="oep_PDLIM1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/PDLIM1_TSNE.png">
</div>
</br>
</br>
<p>Plot tSNE co-expression plots</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">gene_pairs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"TFAP2E"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"TFAP2E"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">))</span><span class="w">
</span><span class="n">lapply</span><span class="p">(</span><span class="n">gene_pairs</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="n">plot_tsne_coexpression</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">gene1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">gene2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w">
</span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">pca</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">22</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">)})</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'FOXI3_LMX1A')">FOXI3/LMX1A</button>
  <button class="tablinks" onclick="openTab(event, 'TFAP2E_LMX1A')">TFAP2E/LMX1A</button>
  <button class="tablinks" onclick="openTab(event, 'FOXI3_SOX8')">FOXI3/SOX8</button>
  <button class="tablinks" onclick="openTab(event, 'TFAP2E_SOX8')">TFAP2E/SOX8</button>
</div>
</br>
<div id="FOXI3_LMX1A" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/FOXI3_LMX1A_co-expression_TSNE.png">
</div>
<div id="TFAP2E_LMX1A" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/TFAP2E_LMX1A_co-expression_TSNE.png">
</div>
<div id="FOXI3_SOX8" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/FOXI3_SOX8_co-expression_TSNE.png">
</div>
<div id="TFAP2E_SOX8" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/oep_subset/tsne/TFAP2E_SOX8_co-expression_TSNE.png">
</div>
</details>
<hr />
</br>
<h3 id="pseudotime-using-a-hrefhttpcole-trapnell-labgithubiomonocle-releasedocsmonocle2a">Pseudotime using <a href="http://cole-trapnell-lab.github.io/monocle-release/docs/">Monocle2</a></h3>
<details><summary class="box">Expand</summary>
<p>
<p>In order to study the process of differentiation from OEP to Otic and Epibranchial lineages, we order the cells in pseudotime using our subset gene modules identified as important for this process.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="c1"># gene modules for biological process of interest</span><span class="w">
</span><span class="n">monocle.input_dims</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">)</span><span class="w">

</span><span class="c1"># input Antler data into Monocle</span><span class="w">
</span><span class="n">df_pheno</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="w">
</span><span class="n">df_feature</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cbind</span><span class="p">(</span><span class="n">fData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">),</span><span class="w"> </span><span class="s1">'gene_short_name'</span><span class="o">=</span><span class="n">fData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">current_gene_names</span><span class="p">)</span><span class="w"> </span><span class="c1"># "gene_short_name" may be required by monocle</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">df_feature</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df_feature</span><span class="o">$</span><span class="n">gene_short_name</span><span class="w">

</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">monocle</span><span class="o">::</span><span class="n">newCellDataSet</span><span class="p">(</span><span class="w">
  </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)),</span><span class="w">
  </span><span class="n">phenoData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">new</span><span class="p">(</span><span class="s2">"AnnotatedDataFrame"</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">df_pheno</span><span class="p">),</span><span class="w">
  </span><span class="n">featureData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">new</span><span class="p">(</span><span class="s1">'AnnotatedDataFrame'</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">df_feature</span><span class="p">),</span><span class="w">
  </span><span class="n">lowerDetectionLimit</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">.1</span><span class="p">,</span><span class="w">
  </span><span class="n">expressionFamily</span><span class="o">=</span><span class="n">VGAM</span><span class="o">::</span><span class="n">tobit</span><span class="p">()</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Dimensionality reduction and order cells along pseudotime</span><span class="w">
</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">estimateSizeFactors</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)</span><span class="w">
</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setOrderingFilter</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">monocle.input_dims</span><span class="p">)</span><span class="w">
</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reduceDimension</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">max_components</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'DDRTree'</span><span class="p">)</span><span class="w">
</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">orderCells</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)</span><span class="w">

</span><span class="c1"># Root cells to earliest "State" (ie DDRTree branch) - which is stage 8</span><span class="w">
</span><span class="n">HSMM</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">orderCells</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">root_state</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">which.max</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)</span><span class="o">$</span><span class="n">State</span><span class="p">,</span><span class="w"> </span><span class="n">pData</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">)[,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">]))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot gradient gene expression on Monocle embeddings</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/gradient_plots/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="k">for</span><span class="p">(</span><span class="n">gn</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">gene_list</span><span class="p">){</span><span class="w">
  </span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_gradient_"</span><span class="p">,</span><span class="w"> </span><span class="n">gn</span><span class="p">,</span><span class="w"> </span><span class="s1">'.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data.frame</span><span class="p">(</span><span class="s1">'Component 1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">1</span><span class="p">,],</span><span class="w"> </span><span class="s1">'Component 2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">2</span><span class="p">,],</span><span class="w">
                          </span><span class="s1">'colour_by'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="m">1+100</span><span class="o">*</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,])</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">log10</span><span class="p">(</span><span class="m">1</span><span class="o">+</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gn</span><span class="p">,]))),</span><span class="w">
                          </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">`Component 1`</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">`Component 2`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">colour_by</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"darkmagenta"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w"> </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w"> </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">ggtitle</span><span class="p">(</span><span class="n">gn</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
          </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)))</span><span class="w">
  </span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_GOI.zip" download>Download
all Monocle gradient plots</a></p>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'monocle_SOX8')">Sox2</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_PAX2')">Pax2</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_TFAP2E')">TFAP2E</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_SIX1')">Six1</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_ZBTB16')">ZBTB16</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_FOXG1')">FOXG1</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_PDLIM1')">PDLIM1</button>
</div>
</br>
<div id="monocle_SOX8" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_SOX8.png">
</div>
<div id="monocle_PAX2" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_PAX2.png">
</div>
<div id="monocle_TFAP2E" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_TFAP2E.png">
</div>
<div id="monocle_SIX1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_SIX1.png">
</div>
<div id="monocle_ZBTB16" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_ZBTB16.png">
</div>
<div id="monocle_FOXG1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_FOXG1.png">
</div>
<div id="monocle_PDLIM1" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/gradient_plots/monocle_gradient_PDLIM1.png">
</div>
</br>
</br>
<p>Plot Monocle co-expression plots</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/coexpression/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot gradient gene co-expression on monocle embeddings</span><span class="w">
</span><span class="n">gene_pairs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"TFAP2E"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LMX1A"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"TFAP2E"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">))</span><span class="w">
</span><span class="n">lapply</span><span class="p">(</span><span class="n">gene_pairs</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="n">monocle_coexpression_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">monocle_obj</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">gene1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">gene2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w">
                                                          </span><span class="n">plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">22</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">)})</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'monocle_FOXI3_LMX1A')">FOXI3/LMX1A</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_TFAP2E_LMX1A')">TFAP2E/LMX1A</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_FOXI3_SOX8')">FOXI3/SOX8</button>
  <button class="tablinks" onclick="openTab(event, 'monocle_TFAP2E_SOX8')">TFAP2E/SOX8</button>
</div>
</br>
<div id="monocle_FOXI3_LMX1A" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/coexpression/FOXI3_LMX1A_co-expression_monocle.png">
</div>
<div id="monocle_TFAP2E_LMX1A" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/coexpression/TFAP2E_LMX1A_co-expression_monocle.png">
</div>
<div id="monocle_FOXI3_SOX8" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/coexpression/FOXI3_SOX8_co-expression_monocle.png">
</div>
<div id="monocle_TFAP2E_SOX8" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/coexpression/TFAP2E_SOX8_co-expression_monocle.png">
</div>
</br>
</br>
<p>Plot trajectories</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/"</span><span class="p">)</span><span class="w">

</span><span class="n">p1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">plot_cell_trajectory</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">color_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cells_samples"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'#BBBDC1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#6B98E9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#05080D'</span><span class="p">),</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cluster"</span><span class="p">)</span><span class="w">
</span><span class="n">p2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">plot_cell_trajectory</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">color_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Pseudotime"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#008ABF"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#E53F00"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Plot cell pseudotime</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_DDRTree_trajectories.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">gridExtra</span><span class="o">::</span><span class="n">grid.arrange</span><span class="p">(</span><span class="n">grobs</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">p1</span><span class="p">,</span><span class="w"> </span><span class="n">p2</span><span class="p">),</span><span class="w"> </span><span class="n">layout_matrix</span><span class="o">=</span><span class="n">matrix</span><span class="p">(</span><span class="n">seq</span><span class="p">(</span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="nb">T</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">


</span><span class="c1"># Plot cell state</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_DDRTree_State_facet.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">plot_cell_trajectory</span><span class="p">(</span><span class="n">HSMM</span><span class="p">,</span><span class="w"> </span><span class="n">color_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"State"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="w"> </span><span class="s1">'#48d1cc'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#f55f20'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#dda0dd'</span><span class="p">),</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"State"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Monocle_DDRTree_trajectories')">Cell pseudotime</button>
  <button class="tablinks" onclick="openTab(event, 'Monocle_DDRTree_State_facet')">Cell state</button>
</div>
</br>
<div id="Monocle_DDRTree_trajectories" class="tabcontent">
  <img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/Monocle_DDRTree_trajectories.png">
</div>
<div id="Monocle_DDRTree_State_facet" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/Monocle_DDRTree_State_facet.png">
</div>
</br>
</br>
<p>Generate a monocle projection plot for each known gene</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/all_monocle_projections/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="n">genes_sel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sort</span><span class="p">(</span><span class="n">intersect</span><span class="p">(</span><span class="w">
  </span><span class="n">getDispersedGenes</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">),</span><span class="w"> </span><span class="m">-1</span><span class="p">),</span><span class="w">
  </span><span class="n">getHighGenes</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">),</span><span class="w"> </span><span class="n">mean_threshold</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">
</span><span class="p">))</span><span class="w">

</span><span class="n">gene_level</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s2">"Normalized"</span><span class="p">)[</span><span class="n">genes_sel</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="n">.</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">getGeneNames</span><span class="p">()],]</span><span class="w">
</span><span class="n">gene_level.2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">apply</span><span class="p">(</span><span class="nf">log</span><span class="p">(</span><span class="m">.1</span><span class="o">+</span><span class="n">gene_level</span><span class="p">),</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="w">
  </span><span class="n">pc_95</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">quantile</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="m">.95</span><span class="p">)</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">pc_95</span><span class="o">==</span><span class="m">0</span><span class="p">){</span><span class="w">
    </span><span class="n">pc_95</span><span class="o">=</span><span class="m">1</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="o">/</span><span class="n">pc_95</span><span class="w">
  </span><span class="n">x</span><span class="p">[</span><span class="n">x</span><span class="o">&gt;</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w">
  </span><span class="nf">as.integer</span><span class="p">(</span><span class="n">cut</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">breaks</span><span class="o">=</span><span class="m">10</span><span class="p">))</span><span class="w">
</span><span class="p">}))</span><span class="w">

</span><span class="k">for</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">gene_level.2</span><span class="p">)){</span><span class="w">
  </span><span class="n">print</span><span class="p">(</span><span class="n">n</span><span class="p">)</span><span class="w">
  </span><span class="n">pdf</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_DDRTree_projection_'</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="s1">'.pdf'</span><span class="p">))</span><span class="w">
  </span><span class="n">plot</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)),</span><span class="w"> </span><span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span><span class="w"> </span><span class="n">main</span><span class="o">=</span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="o">=</span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="o">=</span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">xaxt</span><span class="o">=</span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">yaxt</span><span class="o">=</span><span class="s1">'n'</span><span class="p">,</span><span class="w"> </span><span class="n">asp</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w">
       </span><span class="n">col</span><span class="o">=</span><span class="n">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"#0464DF"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#FFE800"</span><span class="p">))(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)[</span><span class="n">gene_level.2</span><span class="p">[</span><span class="n">n</span><span class="p">,]]</span><span class="w">
  </span><span class="p">)</span><span class="w">
  </span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="n">system</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">"zip -rj "</span><span class="p">,</span><span class="w"> </span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/all_monocle_projections.zip "</span><span class="p">,</span><span class="w"> </span><span class="n">curr_plot_folder</span><span class="p">))</span><span class="w">
</span><span class="n">unlink</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">force</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/all_monocle_projections.zip" download>Download Monocle plots for all known genes</a></p>
</br>
</br>
<p>Plot dotplot for genes along Monocle branches</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"monocle_plots/"</span><span class="p">)</span><span class="w">

</span><span class="c1"># get cell branch information for dotplot</span><span class="w">
</span><span class="n">cell_branch_data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pData</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[,</span><span class="w"> </span><span class="s2">"State"</span><span class="p">,</span><span class="w"> </span><span class="n">drop</span><span class="o">=</span><span class="nb">F</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tibble</span><span class="o">::</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="s1">'cellname'</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">rename</span><span class="p">(</span><span class="n">branch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">State</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">celltype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">case_when</span><span class="p">(</span><span class="w">
    </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"1"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">,</span><span class="w">
    </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"2"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">,</span><span class="w">
    </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"3"</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s1">'OEP'</span><span class="w">
  </span><span class="p">))</span><span class="w">

</span><span class="c1"># gather data for dotplot</span><span class="w">
</span><span class="n">dotplot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="n">gene_list</span><span class="p">,</span><span class="w"> </span><span class="p">]),</span><span class="w"> </span><span class="n">check.names</span><span class="o">=</span><span class="nb">F</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tibble</span><span class="o">::</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="s1">'cellname'</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tidyr</span><span class="o">::</span><span class="n">gather</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">cellname</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">left_join</span><span class="p">(</span><span class="n">cell_branch_data</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"cellname"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># calculate percentage of cells in each cluster expressing gene</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="s1">'Proportion of Cells Expressing'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="o">/</span><span class="n">n</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># scale data</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">scale</span><span class="p">(</span><span class="n">value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="c1"># calculate mean expression</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="s1">'Scaled Average Expression'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">value</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">distinct</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">ungroup</span><span class="p">()</span><span class="w">

</span><span class="c1"># order genes for dotplot based on divergent expression in otic and NC lineages</span><span class="w">
</span><span class="n">gene_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dotplot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">select</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">`Scaled Average Expression`</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">filter</span><span class="p">(</span><span class="n">celltype</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'Otic'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Epib'</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">tidyr</span><span class="o">::</span><span class="n">pivot_wider</span><span class="p">(</span><span class="n">names_from</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">celltype</span><span class="p">,</span><span class="w">
              </span><span class="n">values_from</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">`Scaled Average Expression`</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">order</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Otic</span><span class="o">-</span><span class="n">Epib</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">arrange</span><span class="p">(</span><span class="o">-</span><span class="n">order</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">pull</span><span class="p">(</span><span class="n">genename</span><span class="p">)</span><span class="w">


</span><span class="c1"># add gene order to dotplot_data</span><span class="w">
</span><span class="n">dotplot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dotplot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">mutate</span><span class="p">(</span><span class="n">genename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_order</span><span class="p">))</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s2">"m_oep_dotplot.png"</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">24</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">dotplot_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">genename</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">celltype</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="n">`Proportion of Cells Expressing`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">`Scaled Average Expression`</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_count</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_size_area</span><span class="p">(</span><span class="n">max_size</span><span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_x_discrete</span><span class="p">(</span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlab</span><span class="p">(</span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ylab</span><span class="p">(</span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey90"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">14</span><span class="p">),</span><span class="w">
        </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">c</span><span class="p">(</span><span class="s2">"#48d1cc"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#dda0dd"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#f55f20"</span><span class="p">),</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'bold'</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">16</span><span class="p">),</span><span class="w">
        </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"bottom"</span><span class="p">,</span><span class="w"> </span><span class="n">legend.box</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"horizontal"</span><span class="p">,</span><span class="w"> </span><span class="n">plot.margin</span><span class="o">=</span><span class="n">unit</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span><span class="s2">"cm"</span><span class="p">),</span><span class="w"> </span><span class="n">legend.text</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">),</span><span class="w"> </span><span class="n">legend.title</span><span class="o">=</span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">12</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
</br>
</br>
<p>Plotting the expression of candidate genes, we can easliy classify the Otic, Epibranchial and OEP branches.</p>
<img class="myImages width_80" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/m_oep_dotplot.png">
</details>
<hr />
</br>
<h3 id="co-expression-analysis">Co-expression analysis</h3>
<details><summary class="box">Expand</summary>
<p>
<p>Here we calculate co-expression of key otic and epibranchial markers along each of the Monocle branches. We then test and find that the level of co-expression Otic/Epibranchial markers is higher in the OEP population relative to Otic and Epibranchial branches. This shows that individual OEP cells are co-expressing markers of both lineages - indicative of a bipotent progenenitor population.</p>
<p>First we determine the list of genes used for coexpression analysis.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">otic</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"SOX10"</span><span class="p">,</span><span class="w"> </span><span class="s2">"SOX8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"HOMER2"</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1A'</span><span class="p">)</span><span class="w">
</span><span class="n">epi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"NELL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"FOXI3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"PDLIM1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TFAP2E"</span><span class="p">)</span><span class="w">

</span></code></pre></div></div>
</br>
<p>For each gene pair we calculate the proportion of cells which express both genes in each Monocle branch.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># generate gene pairs for coexpression</span><span class="w">
</span><span class="n">comb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">combn</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">otic</span><span class="p">,</span><span class="w"> </span><span class="n">epi</span><span class="p">),</span><span class="w"> </span><span class="m">2</span><span class="p">))</span><span class="w">

</span><span class="c1"># use data cell branch data from dotplots</span><span class="w">
</span><span class="n">coexpression_data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data.frame</span><span class="p">()</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">pair</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">comb</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">comb</span><span class="p">[</span><span class="n">pair</span><span class="p">,])</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">otic</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">2</span><span class="p">){</span><span class="n">comparison</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"o-o"</span><span class="p">}</span><span class="k">else</span><span class="w"> </span><span class="k">if</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">comb</span><span class="p">[</span><span class="n">pair</span><span class="p">,])</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">otic</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">){</span><span class="n">comparison</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"o-e"</span><span class="p">}</span><span class="k">else</span><span class="p">{</span><span class="n">comparison</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"e-e"</span><span class="p">}</span><span class="w">
  </span><span class="n">newdat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ldply</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">cell_branch_data</span><span class="o">$</span><span class="n">celltype</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nf">c</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="n">colSums</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="n">comb</span><span class="p">[</span><span class="n">pair</span><span class="p">,</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">comb</span><span class="p">[</span><span class="n">pair</span><span class="p">,</span><span class="m">2</span><span class="p">]),</span><span class="w"> </span><span class="n">cell_branch_data</span><span class="p">[</span><span class="n">cell_branch_data</span><span class="o">$</span><span class="n">celltype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="s1">'cellname'</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">cell_branch_data</span><span class="o">$</span><span class="n">celltype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">y</span><span class="p">),</span><span class="w">
      </span><span class="n">comparison</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="p">)</span><span class="w">
  </span><span class="p">})</span><span class="w">
  </span><span class="n">newdat</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">newdat</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w">
  </span><span class="n">coexpression_data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rbind</span><span class="p">(</span><span class="n">coexpression_data</span><span class="p">,</span><span class="w"> </span><span class="n">newdat</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># rename dataframe columns</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">coexpression_data</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"value"</span><span class="p">,</span><span class="w"> </span><span class="s2">"comparison"</span><span class="p">,</span><span class="w"> </span><span class="s2">"branch"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>We then carry out a non-parametric Kruskal Wallis test to compare the proportions of cells co-expressing pairs of genes in each Monocle branch.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># test co-expression in OEP lineage and epibranchial/otic branches</span><span class="w">
</span><span class="n">coexpression_stats</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">coexpression_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="n">branch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">branch</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">group_split</span><span class="p">(</span><span class="n">comparison</span><span class="p">)</span><span class="w">

</span><span class="nf">names</span><span class="p">(</span><span class="n">coexpression_stats</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">coexpression_stats</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="p">[[</span><span class="s2">"comparison"</span><span class="p">]])))</span><span class="w">

</span><span class="c1"># Non parametric kruskal wallis test</span><span class="w">
</span><span class="n">lapply</span><span class="p">(</span><span class="n">coexpression_stats</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">kruskal.test</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">branch</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">))</span><span class="w">

</span><span class="c1"># Wilcoxon rank sum test</span><span class="w">
</span><span class="n">lapply</span><span class="p">(</span><span class="n">coexpression_stats</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s1">'OEP'</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">wilcox.test</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">branch</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.</span><span class="p">))</span><span class="w">
</span><span class="n">lapply</span><span class="p">(</span><span class="n">coexpression_stats</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s1">'OEP'</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">branch</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">wilcox.test</span><span class="p">(</span><span class="n">value</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">branch</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>We calculate the average proportion of cells co-expressing pairs of genes and plot the output.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># calculate mean value per group and SD for plotting bar plot</span><span class="w">
</span><span class="n">plot_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">coexpression_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">branch</span><span class="p">,</span><span class="w"> </span><span class="n">comparison</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">summarise</span><span class="p">(</span><span class="w">
    </span><span class="s2">"Proportion of Cells Co-expressing"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w">
    </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd</span><span class="p">(</span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">group_split</span><span class="p">(</span><span class="n">comparison</span><span class="p">)</span><span class="w">

</span><span class="c1"># rename dataframes split by comparison</span><span class="w">
</span><span class="nf">names</span><span class="p">(</span><span class="n">plot_dat</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">plot_dat</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="p">[[</span><span class="s2">"comparison"</span><span class="p">]])))</span><span class="w">

</span><span class="c1"># bar plots</span><span class="w">
</span><span class="n">oe_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">plot_dat</span><span class="o">$</span><span class="n">`o-e`</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">branch</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_fill_manual</span><span class="p">(</span><span class="s2">"legend"</span><span class="p">,</span><span class="w"> </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Otic"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#f55f20"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#48d1cc"</span><span class="p">,</span><span class="w"> </span><span class="s2">"OEP"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#dda0dd"</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_errorbar</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">ymin</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">-</span><span class="n">sd</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">+</span><span class="n">sd</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">.2</span><span class="p">,</span><span class="w">
                </span><span class="n">position</span><span class="o">=</span><span class="n">position_dodge</span><span class="p">(</span><span class="m">.9</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"**"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.83</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"***"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.85</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.title.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"none"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"Epib-Otic"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">

</span><span class="n">ee_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">plot_dat</span><span class="o">$</span><span class="n">`e-e`</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">branch</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_fill_manual</span><span class="p">(</span><span class="s2">"legend"</span><span class="p">,</span><span class="w"> </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Otic"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#f55f20"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#48d1cc"</span><span class="p">,</span><span class="w"> </span><span class="s2">"OEP"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#dda0dd"</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_errorbar</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">ymin</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">-</span><span class="n">sd</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">+</span><span class="n">sd</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">.2</span><span class="p">,</span><span class="w">
                </span><span class="n">position</span><span class="o">=</span><span class="n">position_dodge</span><span class="p">(</span><span class="m">.9</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"**"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.83</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"*"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.85</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.text.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.ticks.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.line.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.title.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"none"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"Epib-Epib"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">


</span><span class="n">oo_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">plot_dat</span><span class="o">$</span><span class="n">`o-o`</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">branch</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s1">'identity'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_fill_manual</span><span class="p">(</span><span class="s2">"legend"</span><span class="p">,</span><span class="w"> </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Otic"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#f55f20"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#48d1cc"</span><span class="p">,</span><span class="w"> </span><span class="s2">"OEP"</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"#dda0dd"</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_errorbar</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">ymin</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">-</span><span class="n">sd</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="o">=</span><span class="n">`Proportion of Cells Co-expressing`</span><span class="o">+</span><span class="n">sd</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">.2</span><span class="p">,</span><span class="w">
                </span><span class="n">position</span><span class="o">=</span><span class="n">position_dodge</span><span class="p">(</span><span class="m">.9</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Otic"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ns"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.83</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_signif</span><span class="p">(</span><span class="n">comparisons</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"OEP"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epib"</span><span class="p">)),</span><span class="w"> </span><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"**"</span><span class="p">,</span><span class="w">
              </span><span class="n">y_position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">tip_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.02</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="o">=</span><span class="m">0.4</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.85</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.text.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.ticks.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.line.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">axis.title.x</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
        </span><span class="n">legend.position</span><span class="o">=</span><span class="s2">"none"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"Otic-Otic"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s2">"coexpression_test.png"</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">plot_grid</span><span class="p">(</span><span class="n">oe_plot</span><span class="p">,</span><span class="w"> </span><span class="n">ee_plot</span><span class="p">,</span><span class="w"> </span><span class="n">oo_plot</span><span class="p">,</span><span class="w"> </span><span class="n">align</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"hv"</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">rel_widths</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<p>From these three plots we can see that the co-expression of Epibranchial and Otic genes is significantly higher in the OEP branch relative to Otic and Epibranchial branches. This is not the case with the co-expression of pairs of Otic genes or pairs of Epibranchial genes.</p>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/monocle_plots/coexpression_test.png">
</details>
<hr />
</br>
<h3 id="branched-expression-analysis-modeling-beam">Branched expression analysis modeling (BEAM)</h3>
<details><summary class="box">Expand</summary>
<p>
<p>BEAM is used to identify genes with branch dependent expression.</p>
<p>Select genes with high expression and dispersion, and then use BEAM QC to filter and plot.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">
</span><span class="n">genes_sel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">intersect</span><span class="p">(</span><span class="w">
  </span><span class="n">getDispersedGenes</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">),</span><span class="w"> </span><span class="m">-1</span><span class="p">),</span><span class="w">
  </span><span class="n">getHighGenes</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">getReadcounts</span><span class="p">(</span><span class="s1">'Normalized'</span><span class="p">),</span><span class="w"> </span><span class="n">mean_threshold</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">branch_point_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">BEAM_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">BEAM</span><span class="p">(</span><span class="n">HSMM</span><span class="p">[</span><span class="n">genes_sel</span><span class="p">,</span><span class="w"> </span><span class="p">],</span><span class="w"> </span><span class="n">branch_point</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch_point_id</span><span class="p">,</span><span class="w"> </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">num_cores</span><span class="p">)</span><span class="w">

</span><span class="n">BEAM_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">BEAM_res</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">BEAM_res</span><span class="o">$</span><span class="n">qval</span><span class="p">),]</span><span class="w">
</span><span class="n">BEAM_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">BEAM_res</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"gene_short_name"</span><span class="p">,</span><span class="w"> </span><span class="s2">"pval"</span><span class="p">,</span><span class="w"> </span><span class="s2">"qval"</span><span class="p">)]</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_Beam.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">beam_hm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">plot_genes_branched_heatmap</span><span class="p">(</span><span class="n">HSMM</span><span class="p">[</span><span class="n">row.names</span><span class="p">(</span><span class="n">subset</span><span class="p">(</span><span class="n">BEAM_res</span><span class="p">,</span><span class="w"> </span><span class="n">qval</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">.05</span><span class="p">)),],</span><span class="w">
                                      </span><span class="n">branch_point</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch_point_id</span><span class="p">,</span><span class="w">
                                      </span><span class="n">num_clusters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w">
                                      </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ncores</span><span class="p">,</span><span class="w">
                                      </span><span class="n">use_gene_short_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">return_heatmap</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">branch_colors</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'#dd70dd'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#48d1cc'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#f55f20'</span><span class="p">),</span><span class="w">
                                      </span><span class="n">branch_labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Epib"</span><span class="p">,</span><span class="w"> </span><span class="s1">'Otic'</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># save beam score to file</span><span class="w">
</span><span class="n">write.csv</span><span class="p">(</span><span class="n">BEAM_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">arrange</span><span class="p">(</span><span class="n">pval</span><span class="p">),</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'beam_scores.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="nb">F</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/beam/Monocle_Beam.png" width="40%">
</br>
</br>
<p>Plot BEAM for original favourite genes</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_Beam_knownGenes.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">25</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">beam_hm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">plot_genes_branched_heatmap</span><span class="p">(</span><span class="n">HSMM</span><span class="p">[</span><span class="n">m_oep</span><span class="o">$</span><span class="n">favorite_genes</span><span class="p">,],</span><span class="w">
                                      </span><span class="n">branch_point</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch_point_id</span><span class="p">,</span><span class="w">
                                      </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">
                                      </span><span class="n">use_gene_short_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">return_heatmap</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">branch_colors</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'#dd70dd'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#48d1cc'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#f55f20'</span><span class="p">),</span><span class="w">
                                      </span><span class="n">branch_labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Epib"</span><span class="p">,</span><span class="w"> </span><span class="s1">'Otic'</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/beam/Monocle_Beam_knownGenes.png">
</br>
</br>
<p>Plot BEAM for selected markers</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">beam_gene_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">gene_list</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX13'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GATA3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EPHA4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PRDM1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PRDM12'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETV4'</span><span class="p">)</span><span class="w">

</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_Beam_selGenes.png'</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">16</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">beam_hm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">plot_genes_branched_heatmap</span><span class="p">(</span><span class="n">HSMM</span><span class="p">[</span><span class="n">beam_gene_list</span><span class="p">,],</span><span class="w">
                                      </span><span class="n">branch_point</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">branch_point_id</span><span class="p">,</span><span class="w">
                                      </span><span class="n">cluster_rows</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
                                      </span><span class="n">num_clusters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w">
                                      </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">
                                      </span><span class="n">use_gene_short_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">return_heatmap</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w">
                                      </span><span class="n">branch_colors</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'#dd70dd'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#48d1cc'</span><span class="p">,</span><span class="w"> </span><span class="s1">'#f55f20'</span><span class="p">),</span><span class="w">
                                      </span><span class="n">branch_labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Epib"</span><span class="p">,</span><span class="w"> </span><span class="s1">'Otic'</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<img class="myImages" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/beam/Monocle_Beam_selGenes.png">
</details>
<hr />
</br>
<h3 id="rna-velocity">RNA velocity</h3>
<details><summary class="box">Expand</summary>
<p>
<p>Read in splice counts (loom data) and filter cells and genes remaining in OEP subset.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">curr_plot_folder</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="s2">"velocyto/"</span><span class="p">)</span><span class="w">
</span><span class="n">dir.create</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">)</span><span class="w">

</span><span class="c1"># read in loom data, with ensembl ID as rownames instead of gene name</span><span class="w">
</span><span class="n">velocyto_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">custom_read_loom</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">velocyto_input</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'*.loom'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">))</span><span class="w">

</span><span class="c1"># change cell names in velocyto dat to match antler cell names</span><span class="w">
</span><span class="n">velocyto_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">velocyto_dat</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsub</span><span class="p">(</span><span class="s2">".*:"</span><span class="p">,</span><span class="w"> </span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w">
  </span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsub</span><span class="p">(</span><span class="s2">"\\..*"</span><span class="p">,</span><span class="w"> </span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w">
  </span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unname</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="w">
    </span><span class="n">grepl</span><span class="p">(</span><span class="s2">"ss8-TSS"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="p">),</span><span class="w">
    </span><span class="n">sapply</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"_"</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">z</span><span class="p">){</span><span class="n">paste0</span><span class="p">(</span><span class="n">z</span><span class="p">[[</span><span class="m">2</span><span class="p">]],</span><span class="w"> </span><span class="n">z</span><span class="p">[[</span><span class="m">3</span><span class="p">]])}),</span><span class="w">
    </span><span class="n">strsplit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"_"</span><span class="p">)[[</span><span class="m">1</span><span class="p">]][[</span><span class="m">3</span><span class="p">]])))</span><span class="w">
  </span><span class="n">x</span><span class="w">
</span><span class="p">})</span><span class="w">

</span><span class="c1"># get gene annotations from antler object</span><span class="w">
</span><span class="n">antler.gene.names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="o">@</span><span class="n">featureData</span><span class="o">@</span><span class="n">data</span><span class="w">

</span><span class="c1"># keep only cells in m_oep</span><span class="w">
</span><span class="n">m_oep_velocyto_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">velocyto_dat</span><span class="p">,</span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="n">x</span><span class="p">[,</span><span class="n">colnames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="o">$</span><span class="n">Mansel</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">)]</span><span class="w">
</span><span class="p">})</span><span class="w">

</span><span class="c1"># keep only genes in cleaned antler dataset and rename genes based on antler names</span><span class="w">
</span><span class="n">m_oep_velocyto_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">m_oep_velocyto_dat</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="w">
  </span><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">fData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">,]</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">current_gene_names</span><span class="p">[</span><span class="n">match</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="n">fData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">)]</span><span class="w">
  </span><span class="n">x</span><span class="w">
</span><span class="p">})</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Extract count matrices for spliced, unspliced and spanning reads, and calculate RNA velocity</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># exonic read (spliced) expression matrix</span><span class="w">
</span><span class="n">emat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">m_oep_velocyto_dat</span><span class="o">$</span><span class="n">spliced</span><span class="w">
</span><span class="c1"># intronic read (unspliced) expression matrix</span><span class="w">
</span><span class="n">nmat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">m_oep_velocyto_dat</span><span class="o">$</span><span class="n">unspliced</span><span class="w">
</span><span class="c1"># spanning read (intron+exon) expression matrix</span><span class="w">
</span><span class="n">smat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">m_oep_velocyto_dat</span><span class="o">$</span><span class="n">spanning</span><span class="w">

</span><span class="c1"># calculate cell velocity</span><span class="w">
</span><span class="n">rvel</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene.relative.velocity.estimates</span><span class="p">(</span><span class="n">emat</span><span class="p">,</span><span class="n">nmat</span><span class="p">,</span><span class="n">smat</span><span class="o">=</span><span class="n">smat</span><span class="p">,</span><span class="w"> </span><span class="n">kCells</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">diagonal.quantiles</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">fit.quantile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="n">n.cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ncores</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot RNA velocity on tSNE embeddings for cell clusters</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># get tsne embeddings for m_oep cells</span><span class="w">
</span><span class="n">tsne.embeddings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">tsne_embeddings</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">pca</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w">

</span><span class="c1"># return velocity object to plot with ggplot</span><span class="w">
</span><span class="n">vector_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">show.velocity.on.embedding.cor</span><span class="p">(</span><span class="n">tsne.embeddings</span><span class="p">,</span><span class="w"> </span><span class="n">rvel</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">scale</span><span class="o">=</span><span class="s1">'sqrt'</span><span class="p">,</span><span class="w">
                               </span><span class="n">cex</span><span class="o">=</span><span class="m">1.5</span><span class="p">,</span><span class="w"> </span><span class="n">arrow.scale</span><span class="o">=</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">arrow.lwd</span><span class="o">=</span><span class="m">1.5</span><span class="p">,</span><span class="w"> </span><span class="n">n.cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ncores</span><span class="p">,</span><span class="w">
                               </span><span class="n">show.grid.flow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">min.grid.cell.mass</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span><span class="n">grid.n</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">return.details</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

</span><span class="c1"># plot vector map on clusters</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_subset_velocity_clusters_vector.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="p">[[</span><span class="s1">'Mansel'</span><span class="p">]]</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="o">=</span><span class="n">clust.colors</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">garrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot cell arrows on clusters</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_subset_velocity_clusters_arrows.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="p">[[</span><span class="s1">'Mansel'</span><span class="p">]]</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="o">=</span><span class="n">clust.colors</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">arrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'OEP_subset_velocity_clusters_arrows')">Cell clusters: cell arrows</button>
  <button class="tablinks" onclick="openTab(event, 'OEP_subset_velocity_clusters_vector')">Cell clusters: vector map</button>
</div>
</br>
<div id="OEP_subset_velocity_clusters_arrows" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/OEP_subset_velocity_clusters_arrows.png">
</div>
<div id="OEP_subset_velocity_clusters_vector" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/OEP_subset_velocity_clusters_vector.png">
</div>
</br>
</br>
<p>Plot RNA velocity on tSNE embeddings for developmental stage</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">
</span><span class="c1"># plot vector map on stage</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_subset_velocity_stage_vector.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">garrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot cell arrows on stage</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'OEP_subset_velocity_stage_arrows.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">tsne_plot</span><span class="p">(</span><span class="n">m_oep</span><span class="p">,</span><span class="w"> </span><span class="n">m_oep</span><span class="o">$</span><span class="n">topCorr_DR</span><span class="o">$</span><span class="n">genemodules.selected</span><span class="p">,</span><span class="w"> </span><span class="n">seed</span><span class="o">=</span><span class="n">seed</span><span class="p">,</span><span class="w"> </span><span class="n">colour_by</span><span class="o">=</span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">,</span><span class="w"> </span><span class="n">colours</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">,</span><span class="w"> </span><span class="n">perplexity</span><span class="o">=</span><span class="n">perp</span><span class="p">,</span><span class="w"> </span><span class="n">eta</span><span class="o">=</span><span class="n">eta</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">arrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'OEP_subset_velocity_stage_arrows')">Developmental stage: cell arrows</button>
  <button class="tablinks" onclick="openTab(event, 'OEP_subset_velocity_stage_vector')">Developmental stage: vector map</button>
</div>
</br>
<div id="OEP_subset_velocity_stage_arrows" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/OEP_subset_velocity_stage_arrows.png">
</div>
<div id="OEP_subset_velocity_stage_vector" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/OEP_subset_velocity_stage_vector.png">
</div>
</br>
</br>
<p>Plot RNA velocity on Monocle embeddings for cell clusters</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># return velocity object to plot with ggplot</span><span class="w">
</span><span class="n">vector_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">show.velocity.on.embedding.cor</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)),</span><span class="w"> </span><span class="n">rvel</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">scale</span><span class="o">=</span><span class="s1">'sqrt'</span><span class="p">,</span><span class="w">
                                             </span><span class="n">cex</span><span class="o">=</span><span class="m">1.5</span><span class="p">,</span><span class="w"> </span><span class="n">arrow.scale</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">arrow.lwd</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">cell.border.alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">n.cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ncores</span><span class="p">,</span><span class="w">
                                             </span><span class="n">show.grid.flow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">min.grid.cell.mass</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span><span class="n">grid.n</span><span class="o">=</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">return.details</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot vector map on clusters</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_velocity_clusters_vector.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data.frame</span><span class="p">(</span><span class="s1">'Component 1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">1</span><span class="p">,],</span><span class="w"> </span><span class="s1">'Component 2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">2</span><span class="p">,],</span><span class="w">
                  </span><span class="s1">'colour_by'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="p">[[</span><span class="s1">'Mansel'</span><span class="p">]]</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">),</span><span class="w"> </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">),</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">`Component 1`</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">`Component 2`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">colour_by</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clust.colors</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">garrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w"> </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w"> </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span><span class="c1"># plot cell arrows on clusters</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_velocity_clusters_arrows.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data.frame</span><span class="p">(</span><span class="s1">'Component 1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">1</span><span class="p">,],</span><span class="w"> </span><span class="s1">'Component 2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">2</span><span class="p">,],</span><span class="w">
                  </span><span class="s1">'colour_by'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">cellClusters</span><span class="p">[[</span><span class="s1">'Mansel'</span><span class="p">]]</span><span class="o">$</span><span class="n">cell_ids</span><span class="p">),</span><span class="w"> </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">),</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">`Component 1`</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">`Component 2`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">colour_by</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clust.colors</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">arrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w"> </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w"> </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Clusters'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Monocle_velocity_clusters_arrows')">Cell clusters: cell arrows</button>
  <button class="tablinks" onclick="openTab(event, 'Monocle_velocity_clusters_vector')">Cell clusters: vector map</button>
</div>
</br>
<div id="Monocle_velocity_clusters_arrows" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/Monocle_velocity_clusters_arrows.png">
</div>
<div id="Monocle_velocity_clusters_vector" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/Monocle_velocity_clusters_vector.png">
</div>
</br>
</br>
<p>Plot RNA velocity on Monocle embeddings for developmental stage</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">
</span><span class="c1"># plot vector map on stage</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_velocity_stage_vector.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data.frame</span><span class="p">(</span><span class="s1">'Component 1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">1</span><span class="p">,],</span><span class="w"> </span><span class="s1">'Component 2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">2</span><span class="p">,],</span><span class="w">
                  </span><span class="s1">'colour_by'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">),</span><span class="w"> </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">),</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">`Component 1`</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">`Component 2`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">colour_by</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">garrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w"> </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w"> </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Developmental stage'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">


</span><span class="c1"># plot cell arrows on stage</span><span class="w">
</span><span class="n">png</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">curr_plot_folder</span><span class="p">,</span><span class="w"> </span><span class="s1">'Monocle_velocity_stage_arrows.png'</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Arial'</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">400</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">data.frame</span><span class="p">(</span><span class="s1">'Component 1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">1</span><span class="p">,],</span><span class="w"> </span><span class="s1">'Component 2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reducedDimS</span><span class="p">(</span><span class="n">HSMM</span><span class="p">)[</span><span class="m">2</span><span class="p">,],</span><span class="w">
                  </span><span class="s1">'colour_by'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">m_oep</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">),</span><span class="w"> </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">),</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">`Component 1`</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">`Component 2`</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="n">colour_by</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_segment</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">vector_dat</span><span class="o">$</span><span class="n">arrows</span><span class="p">),</span><span class="w">
               </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x0</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y0</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y1</span><span class="p">),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w">
               </span><span class="n">arrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">arrow</span><span class="p">(</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s2">"points"</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"open"</span><span class="p">),</span><span class="w">
               </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey40"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w"> </span><span class="n">axis.ticks</span><span class="o">=</span><span class="n">element_blank</span><span class="p">(),</span><span class="w"> </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">ggtitle</span><span class="p">(</span><span class="s1">'Developmental stage'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">graphics.off</span><span class="p">()</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Monocle_velocity_stage_arrows')">Developmental stage: cell arrows</button>
  <button class="tablinks" onclick="openTab(event, 'Monocle_velocity_stage_vector')">Developmental stage: vector map</button>
</div>
</br>
<div id="Monocle_velocity_stage_arrows" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/Monocle_velocity_stage_arrows.png">
</div>
<div id="Monocle_velocity_stage_vector" class="tabcontent">
  <img class="myImages width_50" id="myImg" src="/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/velocyto/Monocle_velocity_stage_vector.png">
</div>
</details>
<!-- The Modal -->
<div id="myModal" class="modal">
  <!-- The Close Button -->
<p><span class="close">×</span></p>
  <!-- Modal Content (The Image) -->
  <img class="modal-content" id="img01">
  <!-- Modal Caption (Image Text) -->
  <div id="caption"></div>
</div>

</div>
</div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="/public/js/script.js"></script>
  </body>
</html>
