<!DOCTYPE html>
<html lang="en-us">
  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile" />
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />

  <!-- Enable responsiveness on mobile devices-->
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1.0, maximum-scale=1"
  />

  <title>
     Smartseq2_downstream &middot; Otic reprogramming 
  </title>

  
  <link rel="canonical" href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/downstream/smartseq2_downstream/" />
  

  <!-- CSS -->
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/poole.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/syntax.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/custom.css" />
  <link rel="stylesheet" href="https://alexthiery.github.io/otic-reprogramming/public/css/lanyon.css" />
  <link
    rel="stylesheet"
    href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400"
  />

  <!-- stylesheet for icons -->
  <link
    rel="stylesheet"
    href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css"
  />

  <link rel="shortcut icon" href="https://alexthiery.github.io/otic-reprogramming/public/favicon.ico" />

  <i class="fas fa-egg"></i>

  <!-- RSS -->
  <link
    rel="alternate"
    type="application/rss+xml"
    title="RSS"
    href="https://alexthiery.github.io/otic-reprogramming/https://alexthiery.github.io/otic-reprogramming/atom.xml"
  />
</head>


  <body>
    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" />

<!-- Toggleable sidebar -->
<!-- Sidebar icon elements -->

<div class="sidebar" id="sidebar">
  <div class="sidebar-logo">
    <img src="https://alexthiery.github.io/otic-reprogramming/assets/sidebar_icon.png" />
  </div>

  <div id="contact-list" style="text-align: center">
    
    <a href="https://www.kcl.ac.uk/people/andrea-streit">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-users fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="http://orcid.org/0000-0002-5814-4977">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-align-center fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="mailto:alexthiery1@gmail.com">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-envelope fa-stack-1x"></i>
      </span>
    </a>
     
    <a href="https://github.com/alexthiery/">
      <span class="fa-stack fa-lg">
        <i class="fa fa-square-o fa-stack-2x"></i>
        <i class="fa fa-github-alt fa-stack-1x"></i>
      </span>
    </a>
    
  </div>

  <div class="sidebar-item">
    <p>Documentation for the Buzzi et al. 2021 <a href="https://github.com/alexthiery/otic-reprogramming" target="_blank">Github repository</a>.</p>
  </div>

  <!-- sidebar tabs -->

  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/">Home</a>

  <!-- add tabs automatically from downstream and general collections-->

     
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/quick_start/"
    >Quick start</a
  >
      
  <a class="sidebar-nav-item" href="https://alexthiery.github.io/otic-reprogramming/general/smartseq2_alignment/"
    >SmartSeq2 alignment</a
  >
      
  <li class="sidebar-nav-item">
    
    <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Downstream analysis</a>
    <ul>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/downstream_intro/">Intro</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/lmx1a_downstream/">Lmx1a DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/smartseq2_downstream/">SmartSeq2 scRNAseq</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/sox8_downstream/">Sox8 DEA</a>
      </li>
      
      <li class="sidebar-nav-sub-item">
        <a href="https://alexthiery.github.io/otic-reprogramming/downstream/enhancer_downstream/">Enhancer discovery</a>
      </li>
      
    </ul>
  </li>
  

  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming"
    >GitHub repository</a
  >
  <a class="sidebar-nav-item" href="https://github.com/alexthiery/otic-reprogramming/archive/master.zip"
    >Download</a
  >

  <div class="sidebar-item">
    <p>&copy; 2021. All rights reserved.</p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="https://alexthiery.github.io/otic-reprogramming/" title="Home">Otic reprogramming</a>
            <small>Buzzi et al. 2021</small>
          </h3>
        </div>
      </div>

      <div class="container content"><div class="page">
  <!-- hide auto title generation -->
  <!-- <h1 class="page-title">Smartseq2_downstream</h1> -->
  <h2 id="smartseq2-scrnaseq-analysis">SmartSeq2 scRNAseq analysis</h2>
</br>
<p>Automatic switch for running pipeline through Nextflow or interactively in Rstudio</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">getopt</span><span class="p">)</span><span class="w">
</span><span class="n">spec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="w">
  </span><span class="s1">'runtype'</span><span class="p">,</span><span class="w"> </span><span class="s1">'l'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"character"</span><span class="p">,</span><span class="w">
  </span><span class="s1">'cores'</span><span class="w">   </span><span class="p">,</span><span class="w"> </span><span class="s1">'c'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"integer"</span><span class="p">,</span><span class="w">
  </span><span class="s1">'custom_functions'</span><span class="p">,</span><span class="w"> </span><span class="s1">'m'</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s2">"character"</span><span class="w">
</span><span class="p">),</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="n">opt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">getopt</span><span class="p">(</span><span class="n">spec</span><span class="p">)</span><span class="w">

</span><span class="c1"># Set run location</span><span class="w">
</span><span class="k">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">){</span><span class="w">
  </span><span class="n">cat</span><span class="p">(</span><span class="s1">'No command line arguments provided, user defaults paths are set for running interactively in Rstudio on docker\n'</span><span class="p">)</span><span class="w">
  </span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"user"</span><span class="w">
</span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"user"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--runtype must be either 'user' or 'nextflow'"</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="n">tolower</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"null"</span><span class="p">){</span><span class="w">
      </span><span class="n">stop</span><span class="p">(</span><span class="s2">"--custom_functions path must be specified in process params config"</span><span class="p">)</span><span class="w">
    </span><span class="p">}</span><span class="w">
  </span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Set paths and pipeline parameters. Load data, packages and custom functions.</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">{</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"user"</span><span class="p">){</span><span class="w">

    </span><span class="c1"># load custom functions</span><span class="w">
    </span><span class="n">sapply</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s1">'./NF-downstream_analysis/bin/custom_functions/'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">source</span><span class="p">)</span><span class="w">

    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/NF-downstream_analysis/smartseq_analysis/output/"</span><span class="w">
    </span><span class="n">plot_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/NF-downstream_analysis/smartseq_analysis/output/plots/"</span><span class="w">
    </span><span class="n">merged_counts_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/merged_counts/output/'</span><span class="w">
    </span><span class="n">genome_annotations_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-downstream_analysis/extract_gtf_annotations/'</span><span class="w">
    </span><span class="n">gfp_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/merged_counts/output/'</span><span class="w">
    </span><span class="n">velocyto_input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./output/NF-smartseq2_alignment/velocyto/'</span><span class="w">

    </span><span class="c1"># set cores</span><span class="w">
    </span><span class="n">ncores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="w">

  </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">runtype</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"nextflow"</span><span class="p">){</span><span class="w">
    </span><span class="n">cat</span><span class="p">(</span><span class="s1">'pipeline running through nextflow\n'</span><span class="p">)</span><span class="w">

    </span><span class="c1"># load custom functions</span><span class="w">
    </span><span class="n">sapply</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">custom_functions</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">source</span><span class="p">)</span><span class="w">
    </span><span class="n">output_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/"</span><span class="w">
    </span><span class="n">plot_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"./output/plots/"</span><span class="w">
    </span><span class="n">merged_counts_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">genome_annotations_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">gfp_counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">
    </span><span class="n">velocyto_input</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'./'</span><span class="w">

    </span><span class="c1"># set cores</span><span class="w">
    </span><span class="n">ncores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">cores</span><span class="w">
  </span><span class="p">}</span><span class="w">

  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

  </span><span class="c1"># load required packages</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">Antler</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">velocyto.R</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">monocle</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">plyr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">ggsignif</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">rstatix</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">extrafont</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># set pipeline params</span><span class="w">
</span><span class="n">seed</span><span class="o">=</span><span class="m">1</span><span class="w">
</span><span class="n">perp</span><span class="o">=</span><span class="m">5</span><span class="w">
</span><span class="n">eta</span><span class="o">=</span><span class="m">200</span><span class="w">

</span><span class="cd">#' Stage colors</span><span class="w">
</span><span class="n">stage_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"#BBBDC1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#6B98E9"</span><span class="p">,</span><span class="w"> </span><span class="s2">"#05080D"</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'11'</span><span class="p">,</span><span class="w"> </span><span class="s1">'15'</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Load data and initialise Antler object</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Load and hygienize dataset</span><span class="w">
</span><span class="n">m</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Antler</span><span class="o">$</span><span class="n">new</span><span class="p">(</span><span class="n">plot_folder</span><span class="o">=</span><span class="n">plot_path</span><span class="p">,</span><span class="w"> </span><span class="n">num_cores</span><span class="o">=</span><span class="n">ncores</span><span class="p">)</span><span class="w">

</span><span class="c1"># load in phenoData and assayData from ../dataset -&gt; assayData is count matrix; phenoData is metaData (i.e. replicated, conditions, samples etc)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">loadDataset</span><span class="p">(</span><span class="n">folderpath</span><span class="o">=</span><span class="n">merged_counts_path</span><span class="p">)</span><span class="w">

</span><span class="n">pData</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">cells_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">stage_cols</span><span class="p">[</span><span class="nf">as.character</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">expressionSet</span><span class="p">)</span><span class="o">$</span><span class="n">timepoint</span><span class="p">)]</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Plot pre-QC metrics</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">plotReadcountStats</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Raw"</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">category</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s2">"preQC"</span><span class="p">,</span><span class="w"> </span><span class="n">reads_name</span><span class="o">=</span><span class="s2">"read"</span><span class="p">,</span><span class="w"> </span><span class="n">cat_colors</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Some key genes are not annotated in the GTF - manually add these gene names to the annotations file</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># read in annotations file</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="n">genome_annotations_path</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'*gene_annotations.csv'</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">

</span><span class="c1"># add missing annotations to annotations file</span><span class="w">
</span><span class="n">extra_annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'FOXI3'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000037457'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ATN1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000014554'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX10'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000038767'</span><span class="p">,</span><span class="w">
                      </span><span class="s1">'COL11A1'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000005180'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GRHL2'</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ENSGALG00000037687'</span><span class="p">)</span><span class="w">

</span><span class="c1"># add extra annotations to annotations csv file</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">gtf_annotations</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">extra_annotations</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">extra_annotations</span><span class="p">)[</span><span class="n">extra_annotations</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">x</span><span class="p">],</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="m">2</span><span class="p">]))</span><span class="w">

</span><span class="c1"># label extra MT genes</span><span class="w">
</span><span class="n">MT_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'ND3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CYTB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'COII'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ATP8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ND4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ND4L'</span><span class="p">)</span><span class="w">
</span><span class="n">gtf_annotations</span><span class="p">[</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">MT_genes</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s1">'MT-'</span><span class="p">,</span><span class="w"> </span><span class="n">gtf_annotations</span><span class="p">[</span><span class="n">gtf_annotations</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">MT_genes</span><span class="p">,</span><span class="m">2</span><span class="p">])</span><span class="w">

</span><span class="n">write.csv</span><span class="p">(</span><span class="n">gtf_annotations</span><span class="p">,</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'new_annotations.csv'</span><span class="p">),</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">

</span><span class="c1"># set gene annotations</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">setCurrentGeneNames</span><span class="p">(</span><span class="n">geneID_mapping_file</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span><span class="w"> </span><span class="s1">'new_annotations.csv'</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Add list of key genes to Antler object</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cd">#' Store known genes</span><span class="w">
</span><span class="n">apriori_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="w">
  </span><span class="s1">'DACH1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DLX6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXI3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GATA3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GBX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HESX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LMX1B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SALL4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOHO-1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX9'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Known_otic_genes</span><span class="w">
  </span><span class="s1">'ATN1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'BACH2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CNOT4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CXCL14'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DACH2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FEZ1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXP3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HIPK1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HOMER2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'IRX5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'KLF7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'KREMEN1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LDB1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MSI1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PDLIM4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PLAG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PNOC'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PKNOX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'RERE'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SMOC1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SOX13'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TCF7L2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TEAD3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZBTB16'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZFHX3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZNF384'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZNF385C'</span><span class="p">,</span><span class="w"> </span><span class="c1"># New_otic_TFs</span><span class="w">
  </span><span class="s1">'PRDM1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXI1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NKX2-6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NR2F2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PDLIM1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PHOX2B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SALL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TBX10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2E'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TLX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'VGLL2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Epibranchial_genes</span><span class="w">
  </span><span class="s1">'ZNF423'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CXCR4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAFB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYC'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ZEB2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Neural_Genes</span><span class="w">
  </span><span class="s1">'CD151'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETS2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FGFR4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'OTX2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Pax3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX6'</span><span class="p">,</span><span class="w"> </span><span class="s1">'PAX7'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SIX3'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Non_otic_placode_genes</span><span class="w">
  </span><span class="s1">'FOXD3'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ID2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ID4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MSX1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2B'</span><span class="p">,</span><span class="w"> </span><span class="s1">'TFAP2C'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Neural_Crest_Genes</span><span class="w">
  </span><span class="s1">'GATA2'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Epidermis_genes</span><span class="w">
  </span><span class="s1">'COL11A1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DTX4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'GRHL2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NELL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'OTOL1'</span><span class="p">,</span><span class="w"> </span><span class="c1"># Disease_associated_genes</span><span class="w">
  </span><span class="s1">'ARID3A'</span><span class="p">,</span><span class="w"> </span><span class="s1">'BMP4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'CREBBP'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETV4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ETV5'</span><span class="p">,</span><span class="w"> </span><span class="s1">'EYA4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FOXP4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'FSTL4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HOXA2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'JAG1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LFNG'</span><span class="p">,</span><span class="w"> </span><span class="s1">'LZTS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAFA'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MEIS1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYB'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MYCN'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NFKB1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'NOTCH1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SPRY1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SPRY2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SSTR5'</span><span class="p">,</span><span class="w"> </span><span class="c1"># chen et al. 2017</span><span class="w">
  </span><span class="s1">'TWIST1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ASL1'</span><span class="p">,</span><span class="w"> </span><span class="s1">'MAF'</span><span class="w"> </span><span class="c1"># other_genes</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">favorite_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">sort</span><span class="p">(</span><span class="n">apriori_genes</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
</br>
<h3 id="data-pre-processing-and-qc">Data pre-processing and QC</h3>
<details><summary>Expand</summary>
<p>
<p>Remove gene and cell outliers</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cd">#' Remove outliers genes and cells</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">removeOutliers</span><span class="p">(</span><span class="w"> </span><span class="n">lowread_thres</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5e5</span><span class="p">,</span><span class="w">   </span><span class="c1"># select cells with more than 500000 reads</span><span class="w">
                  </span><span class="n">genesmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span><span class="w">       </span><span class="c1"># select cells expressing more than 1k genes</span><span class="w">
                  </span><span class="n">cellmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w">           </span><span class="c1"># select genes expressed in more than 3 cells)</span><span class="w">
                  </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Raw'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>Remove control cells</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotations</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
  </span><span class="s2">"blank"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'241112'</span><span class="p">,</span><span class="w"> </span><span class="s1">'250184'</span><span class="p">,</span><span class="w"> </span><span class="s1">'265102'</span><span class="p">,</span><span class="w"> </span><span class="s1">'272111'</span><span class="p">,</span><span class="w"> </span><span class="s1">'248185'</span><span class="p">,</span><span class="w"> </span><span class="s1">'274173'</span><span class="p">),</span><span class="w">
  </span><span class="s2">"bulk"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'225110'</span><span class="p">,</span><span class="w"> </span><span class="s1">'251172'</span><span class="p">,</span><span class="w"> </span><span class="s1">'273103'</span><span class="p">,</span><span class="w"> </span><span class="s1">'280110'</span><span class="p">,</span><span class="w"> </span><span class="s1">'235161'</span><span class="p">,</span><span class="w"> </span><span class="s1">'246161'</span><span class="p">),</span><span class="w">
  </span><span class="s2">"human"</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'233111'</span><span class="p">,</span><span class="w"> </span><span class="s1">'249196'</span><span class="p">,</span><span class="w"> </span><span class="s1">'257101'</span><span class="p">,</span><span class="w"> </span><span class="s1">'264112'</span><span class="p">,</span><span class="w"> </span><span class="s1">'233185'</span><span class="p">,</span><span class="w"> </span><span class="s1">'247173'</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">m</span><span class="o">$</span><span class="n">excludeCellsFromIds</span><span class="p">(</span><span class="n">which</span><span class="p">(</span><span class="n">m</span><span class="o">$</span><span class="n">getCellsNames</span><span class="p">()</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">annotations</span><span class="p">)))</span><span class="w">
</span></code></pre></div></div>
<p>Remove cells with more than 6% of mitochondrial read counts</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">removeGenesFromRatio</span><span class="p">(</span><span class="w">
  </span><span class="n">candidate_genes</span><span class="o">=</span><span class="n">grep</span><span class="p">(</span><span class="s1">'^MT-'</span><span class="p">,</span><span class="w"> </span><span class="n">m</span><span class="o">$</span><span class="n">getGeneNames</span><span class="p">(),</span><span class="w"> </span><span class="n">value</span><span class="o">=</span><span class="nb">T</span><span class="p">),</span><span class="w">
  </span><span class="n">threshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.06</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</br>
<p>QC plots</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">plotReadcountStats</span><span class="p">(</span><span class="n">data_status</span><span class="o">=</span><span class="s2">"Raw"</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">category</span><span class="o">=</span><span class="s2">"timepoint"</span><span class="p">,</span><span class="w"> </span><span class="n">basename</span><span class="o">=</span><span class="s2">"postQC"</span><span class="p">,</span><span class="w"> </span><span class="n">reads_name</span><span class="o">=</span><span class="s2">"read"</span><span class="p">,</span><span class="w"> </span><span class="n">cat_colors</span><span class="o">=</span><span class="n">unname</span><span class="p">(</span><span class="n">stage_cols</span><span class="p">))</span><span class="w">

</span></code></pre></div></div>
<div class="tab">
  <button class="tablinks" onclick="openTab(event, 'Cells per stage-pre')">Cells per stage</button>
  <button class="tablinks" onclick="openTab(event, 'Readcounts per cell-pre')">Readcounts per cell</button>
  <button class="tablinks" onclick="openTab(event, 'Genes per cell-pre')">Genes per cell</button>
</div>
<div id="Cells per stage-pre" class="tabcontent">
  <embed src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_cellNumber_timepoint_by_timepoint.pdf" width="500" height="500" type="application/pdf">
</div>
<div id="Readcounts per cell-pre" class="tabcontent">
  <embed src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_counts_timepoint_by_timepoint.pdf" width="500" height="500" type="application/pdf">
</div>
<div id="Genes per cell-pre" class="tabcontent">
  <embed src="https://alexthiery.github.io/otic-reprogramming/assets/output/NF-downstream_analysis/smartseq_analysis/output/plots/preQC_statistics_counts_timepoint_by_timepoint.pdf" width="500" height="500" type="application/pdf">
</div>
</br>
</br>
<p>Normalize readcounts to CPM and remove genes with &lt;10 CPM</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">m</span><span class="o">$</span><span class="n">normalize</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s2">"Count-Per-Million"</span><span class="p">)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">excludeUnexpressedGenes</span><span class="p">(</span><span class="n">min.cells</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">
</span><span class="n">m</span><span class="o">$</span><span class="n">removeLowlyExpressedGenes</span><span class="p">(</span><span class="n">expression_threshold</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">selection_theshold</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">data_status</span><span class="o">=</span><span class="s1">'Normalized'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
</details>

</div>
</div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script src="https://alexthiery.github.io/otic-reprogramming/public/js/script.js"></script>
  </body>
</html>
